<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>`orthos` is an R package for variance decomposition using
    conditional variational auto-encoders • orthos</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="`orthos` is an R package for variance decomposition using
    conditional variational auto-encoders">
<meta property="og:description" content="`orthos` decomposes RNA-seq contrasts, for example obtained from a
    gene knock-out or compound treatment experiment, into unspecific and
    experiment-specific components. Original and decomposed contrasts can
    be efficiently queried against a large database of contrasts
    (derived from ARCHS4, https://maayanlab.cloud/archs4/) to identify similar
    experiments. `orthos` furthermore provides plotting functions to visualize
    the results of such a search for similar contrasts.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">orthos</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/orthosIntro.html">An introduction to contrast decomposition and querying using orthos</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<p><img src="articles/orthos.png" alt="orthos" width="160"></p>
<div class="section level1">
<div class="page-header"><h1 id="orthos---mechanistic-insights-from-differential-gene-expression-experiments-with-deep-generative-networks">
<em>orthos</em> - Mechanistic insights from differential gene expression experiments with Deep Generative Networks<a class="anchor" aria-label="anchor" href="#orthos---mechanistic-insights-from-differential-gene-expression-experiments-with-deep-generative-networks"></a>
</h1></div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>Differential expression analysis upon cellular perturbations is the most commonly used tool for probing molecular mechanisms of action due to its simplicity and low cost. However, interpretation of such experiments is confounded by the complex and nuanced impact of experimental treatments on cellular processes.</p>
<p>Enter <code>orthos</code>, a generative modelling-based approach that disentangles experiment-specific from non-specific effects of perturbations on gene expression. Utilizing a large corpus of gene expression contrasts (&gt;100K annotated / &gt;1M augmented) compiled from the <a href="https://maayanlab.cloud/archs4/" class="external-link">ARCHS4</a> database of uniformly processed RNAseq experiments we trained conditional variational models that accurately capture and isolate “<strong>polytopic</strong>” (non-specific, observed across multiple treatments) effects while accounting for context. The residual, <strong>specific</strong> component obtained from this decomposition offers a more unequivocal experimental signature and is a proxy more closely related to the direct molecular effects of a treatment when compared to the raw signal.</p>
<p>In addition to providing a clearer understanding of the effects of experimental treatments on gene exrpession <code>orthos</code> also enables researchers to query the contrast database with arbitrary contrasts and identify experiments with similar specific effects ultimately helping to map pharmacological, physiological or genetic treatments to mechanisms of action.</p>
<p><code>orthos</code> is developed at the <a href="https://www.fmi.ch/" class="external-link">FMI-Friedrich Miescher Institute for Biomedical Research</a> and is supported by funding from the <a href="https://www.novartis.com/research-development/novartis-institutes-biomedical-research" class="external-link">NIBR-Novartis Institutes for Biomedical Research</a>.</p>
<p>Current contributors include:</p>
<ul>
<li><a href="https://github.com/ppapasaikas" class="external-link">Panagiotis Papasaikas</a></li>
<li><a href="https://github.com/csoneson" class="external-link">Charlotte Soneson</a></li>
<li><a href="https://github.com/mbstadler" class="external-link">Michael Stadler</a></li>
</ul>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p><code>orthos</code> can be installed from from Bioconductor using <code><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">BiocManager::install()</a></code>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"orthos"</span><span class="op">)</span></span>
<span><span class="co"># or also...</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"orthos"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="functionality">Functionality<a class="anchor" aria-label="anchor" href="#functionality"></a>
</h2>
<p>The workhorse behind <code>orthos</code> are organism-specific conditional variational autoencoder (cVAE) models that break down the variance of a given differential expression experiment into a non-specific (polytopic) and an experiment-specific components.</p>
<p>The non-specific (polytopic) component corresponds to gene variance that has been observed across multiple instances during training, while the experiment-specific variance is fairly unique to the experiment.</p>
<p>Given these inputs, the model maps the contrast to a concise latent representation (z<sub>D</sub>) which is then used to generate a decoded version of the contrast.</p>
<p>This decoded contrast corresponds directly to the <strong>polytopic</strong> (non-specific) component of the observed effects; it subsumes the gene variance that the model can account for because it has been encountered, in some form, during training. The residual obtained after removing the decoded contrast from the original one is the specific component; this encompasses the gene variance that the model cannot account for (random noise + experiment-specific biological effects).</p>
<p><img src="articles/orthos_arch.png" alt="orthos" width="600"></p>
<p>A typical analysis with <code>orthos</code> involves two main steps:</p>
<ol style="list-style-type: decimal">
<li><p>Decomposing one or several contrasts into their corresponding specific and polytopic (non-specific) fraction and</p></li>
<li><p>Performing queries with the original and decomposed specific and polytopic fractions against the <code>orthosData</code> contrast database</p></li>
</ol>
<p>For more details on usage please refer to the package vignette.</p>
<div class="section level3">
<h3 id="miscellaneous">Miscellaneous<a class="anchor" aria-label="anchor" href="#miscellaneous"></a>
</h3>
</div>
<div class="section level3">
<h3 id="code-of-conduct">Code of Conduct<a class="anchor" aria-label="anchor" href="#code-of-conduct"></a>
</h3>
</div>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/orthos" class="external-link">View on Bioconductor</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing orthos</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Panagiotis Papasaikas <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-1640-7636" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>
<a href="https://csoneson.github.io" class="external-link">Charlotte Soneson</a> <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-3833-2169" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>
<a href="https://www.fmi.ch/platforms/platform.html?plt=115" class="external-link">Michael Stadler</a> <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-2269-4934" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Panagiotis Papasaikas, <a href="https://csoneson.github.io" class="external-link">Charlotte Soneson</a>, <a href="https://www.fmi.ch/platforms/platform.html?plt=115" class="external-link">Michael Stadler</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
