<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Decompose input contrasts to decoded and residual fractions — decomposeVar • orthos</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Decompose input contrasts to decoded and residual fractions — decomposeVar"><meta property="og:description" content="Decompose input contrasts (gene expression deltas) to decoded (generic)
and residual (unique) components according to a contrast encoder-decoder
pre-trained on a large corpus of public RNAseq experiments."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthos</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/orthosIntro.html">An introduction to contrast decomposition and querying using orthos</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Decompose input contrasts to decoded and residual fractions</h1>
    
    <div class="hidden name"><code>decomposeVar.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Decompose input contrasts (gene expression deltas) to decoded (generic)
and residual (unique) components according to a contrast encoder-decoder
pre-trained on a large corpus of public RNAseq experiments.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">decomposeVar</span><span class="op">(</span></span>
<span>  <span class="va">M</span>,</span>
<span>  MD <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  treatm <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cntr <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  processInput <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  organism <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Human"</span>, <span class="st">"Mouse"</span><span class="op">)</span>,</span>
<span>  featureType <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AUTO"</span>, <span class="st">"ENSEMBL_GENE_ID"</span>, <span class="st">"GENE_SYMBOL"</span>, <span class="st">"ENTREZ_GENE_ID"</span>,</span>
<span>    <span class="st">"ARCHS4_ID"</span><span class="op">)</span>,</span>
<span>  pseudocount <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>M</dt>
<dd><p>Matrix of raw gene counts.</p></dd>


<dt>MD</dt>
<dd><p>Matrix of gene deltas (optional). If <code>MD</code> is specified,
<code>M</code> is assumed to be a raw gene count matrix specifying context for
contrasts specified in <code>MD</code>. <code>MD</code> is then a matrix of gene
deltas with the same dimensions as <code>M</code>. If <code>MD</code> is specified,
<code>treatm</code> and <code>cntr</code> have to be <code>NULL</code>.</p></dd>


<dt>treatm, cntr</dt>
<dd><p>Vectors indicating column indices in <code>M</code>
corresponding to treatments and controls. If <code>treatm</code> and
<code>cntr</code> are specified, <code>MD</code> has to be <code>NULL</code>.</p></dd>


<dt>processInput</dt>
<dd><p>If set to <code>TRUE</code> (default) the count matrix will
be preprocessed (library normalized, log2-transformed after addition of
a pseudocount, NA values will be set to 0).</p></dd>


<dt>organism</dt>
<dd><p>Selects the autoencoder model trained on data from this
species. One of <code>"Human"</code> or <code>"Mouse"</code>.</p></dd>


<dt>featureType</dt>
<dd><p>Set to <code>"AUTO"</code> for automatic feature id-type
detection. Alternatively specify the type of supplied id features.
Current supported types are <code>"ENSEMBL_GENE_ID"</code>,
<code>"GENE_SYMBOL"</code>, <code>"ENTREZ_GENE_ID"</code> and <code>"ARCHS4_ID"</code>.</p></dd>


<dt>pseudocount</dt>
<dd><p>Numerical scalar, added to raw counts in <code>M</code> when
<code>preprocessInput = TRUE</code>.</p></dd>


<dt>verbose</dt>
<dd><p>Logical scalar indicating whether to print messages along
the way.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></code> object
    with the decomposed contrasts in the assays and the decomposed variance
    as the <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></code>.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>When calling <code>decomposeVar()</code>, you may see an <code>ImportError</code>
    on the console. This most likely does not have any negative consequences,
    rather it means that R and python may not be library compatible and that
    an automated fallback approach is being used (for more details see
    <code>testload</code> argument of <code><a href="https://rdrr.io/pkg/basilisk/man/basiliskStart.html" class="external-link">basiliskStart</a></code>).</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Panagiotis Papasaikas</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">MKL1_human</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GSE215150_MKL1_Human.rds"</span>,</span></span>
<span class="r-in"><span>package <span class="op">=</span> <span class="st">"orthos"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Specifying M, treatm and cntr:</span></span></span>
<span class="r-in"><span><span class="va">dec_MKL1_human</span> <span class="op">&lt;-</span> <span class="fu">decomposeVar</span><span class="op">(</span>M <span class="op">=</span> <span class="va">MKL1_human</span>, treatm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span>, cntr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, </span></span>
<span class="r-in"><span>                              organism <span class="op">=</span> <span class="st">"Human"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?orthosData and browseVignettes('orthosData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> require(“keras”)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?orthosData and browseVignettes('orthosData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?orthosData and browseVignettes('orthosData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"><span>                              </span></span>
<span class="r-in"><span>                              </span></span>
<span class="r-in"><span><span class="co"># Alternatively by specifying M and MD:</span></span></span>
<span class="r-in"><span><span class="va">pseudocount</span> <span class="op">&lt;-</span> <span class="fl">4</span> </span></span>
<span class="r-in"><span><span class="va">M</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sweep.html" class="external-link">sweep</a></span><span class="op">(</span><span class="va">MKL1_human</span>, <span class="fl">2</span>,</span></span>
<span class="r-in"><span>            <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">MKL1_human</span><span class="op">)</span>, FUN <span class="op">=</span> <span class="st">"/"</span><span class="op">)</span> <span class="op">*</span> <span class="fl">1e+06</span></span></span>
<span class="r-in"><span><span class="va">M</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">M</span> <span class="op">+</span> <span class="va">pseudocount</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">DeltaM</span> <span class="op">&lt;-</span> <span class="va">M</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MKL1"</span>,<span class="st">"caMKL1"</span><span class="op">)</span><span class="op">]</span><span class="op">-</span><span class="va">M</span><span class="op">[</span>,<span class="st">"Ctrl"</span><span class="op">]</span> <span class="co"># Matrix of contrasts</span></span></span>
<span class="r-in"><span><span class="va">ContextM</span> <span class="op">&lt;-</span> <span class="va">M</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Ctrl"</span>,<span class="st">"Ctrl"</span><span class="op">)</span><span class="op">]</span> <span class="co"># Matrix with context for the specified contrasts</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">ContextM</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">DeltaM</span><span class="op">)</span> <span class="co"># M and MD need identical dimnames                       </span></span></span>
<span class="r-in"><span><span class="va">RES</span> <span class="op">&lt;-</span> <span class="fu">decomposeVar</span><span class="op">(</span>M <span class="op">=</span> <span class="va">ContextM</span>, MD <span class="op">=</span> <span class="va">DeltaM</span>, processInput <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Checking input...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> demo_decomposed_contrasts_human_rds  already present in cache at: /Users/runner/Library/Caches/org.R-project.R/R/ExperimentHub/human_v212_NDF_c100_DEMOse.rds</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> demo_decomposed_contrasts_human_hdf5  already present in cache at: /Users/runner/Library/Caches/org.R-project.R/R/ExperimentHub/human_v212_NDF_c100_DEMOassays.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Detecting feature ids-type...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Feature ids-type detected: GENE_SYMBOL</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 18079/59453 provided input features mapped against a total of 20411 model features.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2332 missing features will be set to 0.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --&gt; Missing features corresponding to non/lowly expressed genes in your context(s) are of no consequence.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --&gt; The model is robust to small fractions (&lt;10%) of missing genes that are expressed in your context(s).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> --&gt; Increased numbers of missing expressed genes in your input might result in model performance decline.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encoding context...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?orthosData and browseVignettes('orthosData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Encoding and decoding contrasts...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?orthosData and browseVignettes('orthosData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> see ?orthosData and browseVignettes('orthosData') for documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing output...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Panagiotis Papasaikas, <a href="https://csoneson.github.io" class="external-link">Charlotte Soneson</a>, <a href="https://www.fmi.ch/platforms/platform.html?plt=115" class="external-link">Michael Stadler</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

