<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An introduction to contrast decomposition and querying using orthos • orthos</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An introduction to contrast decomposition and querying using orthos">
<meta property="og:description" content="orthos">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthos</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/orthosIntro.html">An introduction to contrast decomposition and querying using orthos</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>An introduction to contrast decomposition and
querying using orthos</h1>
                        <h4 data-toc-skip class="author">Panagiotis
Papasaikas</h4>
            <address class="author_afil">
      Friedrich Miescher Institute for Biomedical Research, Basel,
SwitzerlandSIB Swiss Institute of
Bioinformatics<br><a class="author_email" href="mailto:#"></a><a href="mailto:panp80@gmail.com" class="email">panp80@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Charlotte
Soneson</h4>
            <address class="author_afil">
      Friedrich Miescher Institute for Biomedical Research, Basel,
SwitzerlandSIB Swiss Institute of
Bioinformatics<br><h4 data-toc-skip class="author">Michael
Stadler</h4>
            <address class="author_afil">
      Friedrich Miescher Institute for Biomedical Research, Basel,
SwitzerlandSIB Swiss Institute of Bioinformatics<br><h4 data-toc-skip class="date">28 April 2023</h4>
      
      
      <div class="hidden name"><code>orthosIntro.Rmd</code></div>

    </address>
</address>
</div>

    
    
<p><img src="../../../_temp/Library/orthos/orthos.png" width="50%" style="display: block; margin: auto;"></p>
<style type="text/css">
body {
  margin: 0 auto;
  max-width: 1600px;
  padding: 2rem;
}
</style>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>RNAseq-based differential expression analysis upon cellular
perturbations, such as gene knockouts, RNA knockdowns or compound
treatment experiments, is the most commonly used tool for probing
molecular mechanisms of action due to its simplicity and low cost.</p>
<p>However, interpretation of such gene expression contrasts is
confounded by the complex and nuanced impacts of experimental treatments
on cellular processes.</p>
<p>For example, knockout or over-expression of a transcription factor
will not only alter the transcription of its direct target genes, but
also cause many secondary expression changes. In addition, treatments or
treatment delivery agents typically elicit a variety of unintended,
systemic responses (such as immune, toxic, metabolic) that cannot be
well-controlled for by the design of the study.</p>
<p>The final experimentally measured gene expression changes are a hard
to assess convolution of specific and non-specific secondary and lateral
treatment effects.</p>
<p><code>orthos</code> is a generative modelling-based approach that
disentangles the experiment-specific from the non-specific effects of
perturbations on gene expression. It is trained on a large corpus of
gene expression contrasts (per organism &gt;60K annotated, &gt;0.5M
augmented), compiled from the <a href="https://maayanlab.cloud/archs4/" class="external-link">ARCHS4</a> database of uniformly
processed RNAseq experiments (<span class="citation">Lachmann et al.
(2018)</span>). It accurately captures and isolates
<strong>polytopic</strong> effects (non-specific effects that are
observed across multiple treatments) while accounting for context
(tissue or cell-line experimental background).</p>
<p>The residual <strong>specific</strong> component obtained from this
decomposition offers a more unequivocal experimental signature and is
more closely related to the direct molecular effects of the perturbation
when compared to the raw signal.</p>
<p>In addition to providing a clearer understanding of the effects of
experimental treatments on gene expression, <code>orthos</code> also
enables researchers to query the contrast database with arbitrary
contrasts and identify experiments with similar specific effects,
ultimately helping to map treatments to mechanisms of action.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>After installation, the package can be loaded with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">orthos</span><span class="op">)</span></span></code></pre></div>
<p>A typical analysis involves two steps:</p>
<ol style="list-style-type: decimal">
<li><p>Decomposing one or several contrasts into their corresponding
specific and polytopic components using the <code><a href="../reference/decomposeVar.html">decomposeVar()</a></code>
function and</p></li>
<li><p>Performing a query with the original and decomposed specific and
polytopic contrasts against the contrast database using the
<code><a href="../reference/queryWithContrasts.html">queryWithContrasts()</a></code> function.</p></li>
</ol>
</div>
<div class="section level2">
<h2 id="demonstration-data">Demonstration data<a class="anchor" aria-label="anchor" href="#demonstration-data"></a>
</h2>
<p>To demonstrate the functionality of <code>orthos</code> we use a
dataset from the the GEO series <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE215150" class="external-link">GSE215150</a>.</p>
<p>This series was not part of the <code>orthos</code> training or the
<code>orthosData</code> contrast database; it was only publicly released
on January 1st 2023 after the freeze of the training data to the ARCHS4
v2.1.2 database.</p>
<p>The performed experiment involves over-expression (OE) of the
MKL/megakaryoblastic leukemia 1 gene (also termed MRTFA/myocardin
related transcription factor A) and a constitutively active mutant MKL1
(caMKL1, described in <span class="citation">Hu et al. (2019)</span>).
Both OE experiments were performed in mouse LM2 and human 4T1
tumor-derived breast cancer cell lines. In addition to the MKL1/caMKL1
OE samples, the series also contains no-treatment controls for each of
the two cell lines.</p>
<p>For simplicity the three biological replicates of each available
condition have been collapsed in the data provided in the package.</p>
<p>In the provided form each of the two datasets (Mouse, Human) contains
raw counts for over 55,000 genes identified by gene symbols in three
conditions: Control (Ctrl), MKL1 OE (MKL1) and constitutively-active
MKL1 OE (caMKL1):</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MKL1_human</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GSE215150_MKL1_Human.rds"</span>,</span>
<span>                                  package <span class="op">=</span> <span class="st">"orthos"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">MKL1_human</span><span class="op">)</span></span>
<span><span class="co">#&gt;             Ctrl MKL1 caMKL1</span></span>
<span><span class="co">#&gt; DDX11L1        6    3      4</span></span>
<span><span class="co">#&gt; WASH7P        47   57     41</span></span>
<span><span class="co">#&gt; MIR6859-1     10   10      6</span></span>
<span><span class="co">#&gt; MIR1302-2HG    0    6      1</span></span>
<span><span class="co">#&gt; MIR1302-2      0    0      0</span></span>
<span><span class="co">#&gt; FAM138A        1    1      5</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">MKL1_human</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 59453     3</span></span>
<span><span class="va">MKL1_mouse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GSE215150_MKL1_Mouse.rds"</span>,</span>
<span>                                  package <span class="op">=</span> <span class="st">"orthos"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">MKL1_mouse</span><span class="op">)</span></span>
<span><span class="co">#&gt;               Ctrl MKL1 caMKL1</span></span>
<span><span class="co">#&gt; 4933401J01Rik    0    0      0</span></span>
<span><span class="co">#&gt; Gm26206          0    0      0</span></span>
<span><span class="co">#&gt; Xkr4             0    0      0</span></span>
<span><span class="co">#&gt; Gm18956          0    0      0</span></span>
<span><span class="co">#&gt; Gm37180          0    0      0</span></span>
<span><span class="co">#&gt; Gm37363          0    0      0</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">MKL1_mouse</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 55367     3</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="decomposition-of-differential-gene-expression-variance-into-specific-and-non-specific-components-using-decomposevar">Decomposition of differential gene expression variance into specific
and non-specific components using <code>decomposeVar()</code><a class="anchor" aria-label="anchor" href="#decomposition-of-differential-gene-expression-variance-into-specific-and-non-specific-components-using-decomposevar"></a>
</h2>
<p>The workhorse behind <code>orthos</code> are organism-specific
conditional variational autoencoder (cVAE) models that break down the
variance of a given differential expression experiment into a
non-specific (polytopic) and an experiment-specific component.</p>
<p>The non-specific (polytopic) component corresponds to gene variance
that has been observed across multiple instances during training, while
the experiment-specific variance is fairly unique to the experiment.</p>
<p>The inputs to the models are gene counts in the form of
log2-transformed counts per million (LCPMs) that are used to encode the
<strong>context</strong> of the performed experiment as well as the
actual gene expression contrasts in the form of log2 fold-changes
(LFCs), i.e log2-transformed CPM ratios. As we will see, calculation of
those inputs is by default performed internally given only <strong>raw
gene counts</strong> and a specification of the contrasted
conditions.</p>
<p>Given these inputs, the model maps the contrast to a concise latent
representation (z<sub>D</sub>) which is then used to generate a decoded
version of the contrast.</p>
<p>This decoded contrast corresponds directly to the
<strong>polytopic</strong> (non-specific) component of the observed
effects; it subsumes the gene variance that the model can account for
because it has been encountered, in some form, during training. The
residual obtained after removing the decoded contrast from the original
one is the specific component; this encompasses the gene variance that
the model cannot account for (experiment-specific biological effects +
random noise).</p>
<div class="figure">
<img src="orthos_arch.png" style="width:100.0%" alt=""><p class="caption">orthos architecture</p>
</div>
<p>The function that performs this contrast decomposition into the
polytopic and specific components is <code><a href="../reference/decomposeVar.html">decomposeVar()</a></code>. There
are two available modes in which the required inputs can be fed into the
function:</p>
<ul>
<li><p>In the first mode the user passes the matrix <code>M</code> of
raw counts (genes in rows, conditions in columns) and two vectors
<code>treatm</code> and <code>cntr</code> of equal length specifying the
column indices in <code>M</code> corresponding to the “treatments” and
their respective “controls”. The same column indices can be repeated
multiple times in these vectors, for example in the case where multiple
treatments are paired to the same control.</p></li>
<li><p>In the second mode the user passes the matrix <code>M</code> of
raw counts and a second matrix <code>MD</code> that contains
precalculated log2 fold-changes for the contrasts to be analyzed. In
this case, <code>M</code> specifies the context for the corresponding
columns of <code>MD</code>, and thus the two matrices need to have the
same dimensionality and identical row- and column-names.</p></li>
</ul>
<p>In both modes the rownames of <code>M</code> (and <code>MD</code> if
specified) need to correspond to valid gene identifiers (currently
<code>orthos</code> supports Entrez gene identifiers, ENSEMBL gene
identifiers, gene symbols or ARCHS4 gene identifiers). By default the
type of gene identifier is detected automatically.</p>
<p>The first time that <code>decomposeVar</code> is executed for a
particular organism, the models required for inference will be
automatically downloaded from <code>ExperimentHub</code> and cached in
the user ExperimentHub directory (see
<code>ExperimentHub::getExperimentHubOption("CACHE")</code>) using the
<code>orthosData</code> companion package.</p>
<p>For the MKL1 data, typical calls of <code><a href="../reference/decomposeVar.html">decomposeVar()</a></code> using
the first mode would look like this:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Decompose MKL1-vs-Cntrl and caMKL1-vs-Cntrl contrasts for human:</span></span>
<span><span class="va">dec_MKL1_human</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/decomposeVar.html">decomposeVar</a></span><span class="op">(</span>M <span class="op">=</span> <span class="va">MKL1_human</span>, treatm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span>, cntr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, </span>
<span>                               organism <span class="op">=</span> <span class="st">"Human"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">dec_MKL1_human</span></span>
<span></span>
<span><span class="co">#Decompose MKL1-vs-Cntrl and caMKL1-vs-Cntrl contrasts for mouse:</span></span>
<span><span class="va">dec_MKL1_mouse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/decomposeVar.html">decomposeVar</a></span><span class="op">(</span>M <span class="op">=</span> <span class="va">MKL1_mouse</span>, treatm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span>, cntr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>                               organism <span class="op">=</span> <span class="st">"Mouse"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">dec_MKL1_mouse</span></span></code></pre></div>
<p>The output of <code><a href="../reference/decomposeVar.html">decomposeVar()</a></code> is a
<code>SummarizedExperiment</code> object with dimensions <code>N</code>
x <code>M</code>, where <code>N</code> is the number of
<code>orthos</code> genes for that organism and <code>M</code> is the
number of contrasts specified during input.</p>
<p>Notice that, of the total gene features present in the input (over
55,000), only those that are part of the <code>orthos</code> model
(~20,000 genes) are used.</p>
<p>These ~20,000 <code>orthos</code> genes are “sanctioned” according to
several criteria (located on canonical chromosomes, no pseudogenes, no
ribosomal protein genes, detected in at least a small fraction of the
ARCHS4 database).</p>
<p>The model is highly robust to small fractions of <code>orthos</code>
genes not being part of the user input, even if those genes are
expressed in the context under consideration. That being noted, it is
safer to feed-in inputs that are as comprehensive as possible (i.e not
filtered in any way) in terms of gene features.</p>
<p>The <code>SummarizedExperiment</code> output also has 4 assay slots
corresponding to the input contrasts, decoded (polytopic), and residual
(specific) components, as well as the gene context. Contrasts are
represented as log2 fold-changes (LFCs) and context is represented as
log2-transformed counts per million (log2 CPM).</p>
<p>The <code>colData</code> of the object summarizes the proportion of
variance accounted for in each decomposed component:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">dec_MKL1_human</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#MA plot of for the input contrasts:</span></span>
<span><span class="va">DF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>L2CPM<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">dec_MKL1_human</span>,<span class="st">"CONTEXT"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>,</span>
<span>                 L2FC<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">dec_MKL1_human</span>,<span class="st">"INPUT_CONTRASTS"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span>data<span class="op">=</span><span class="va">DF</span>, <span class="fu">aes</span><span class="op">(</span>x<span class="op">=</span><span class="va">L2CPM</span>, y<span class="op">=</span><span class="va">L2FC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_point</span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.4</span>, size<span class="op">=</span><span class="fl">1.8</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"darkgray"</span>, linewidth <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Expression (Log2 CPMs)"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ylab</span><span class="op">(</span><span class="st">"Log2 Fold Change"</span><span class="op">)</span>  </span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">dec_MKL1_human</span><span class="op">)</span> </span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">dec_MKL1_mouse</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">dec_MKL1_mouse</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="querying-the-database-of-gene-contrasts-using-querywithcontrasts">Querying the database of gene contrasts using
<code>queryWithContrasts()</code><a class="anchor" aria-label="anchor" href="#querying-the-database-of-gene-contrasts-using-querywithcontrasts"></a>
</h2>
<p>Typically, the next step of the analysis involves querying the
contrasts database (<code>orthosData</code>) to identify public
experiments similar to the one(s) under investigation, either in terms
of the original or decomposed (polytopic/ specific) contrasts.</p>
<p><code>orthosData</code> contains over 100,000 differential gene
expression experiments compiled from the ARCHS4 database of publicly
available expression data (<span class="citation">Lachmann et al.
(2018)</span>). Each entry in <code>orthosData</code> corresponds to a
pair of RNAseq samples contrasting a treatment vs a control condition. A
combination of metadata, semantic and quantitative analyses was used to
determine the proper assignment of samples to such pairs in
<code>orthosData</code>.</p>
<p>The function that performs the queries against
<code>orthosData</code> is <code><a href="../reference/queryWithContrasts.html">queryWithContrasts()</a></code>. The input
to this function is the <code>SummarizedExperiment</code> object
obtained in the previous step from <code><a href="../reference/decomposeVar.html">decomposeVar()</a></code>, either
the complete object or one that has been column-subsetted, allowing to
query the contrast database with only a subset of the decomposed
contrasts.</p>
<p>As was the case for the <code>orthos</code> models, a database will
be automatically downloaded from <code>ExperimentHub</code> and cached
in the user ExperimentHub directory (see
<code>ExperimentHub::getExperimentHubOption("CACHE")</code>) using the
<code>orthosData</code> companion package, the first time
<code><a href="../reference/queryWithContrasts.html">queryWithContrasts()</a></code> is called for that database or the
first time the user attempts to access the database directly with
<code><a href="../reference/loadContrastDatabase.html">loadContrastDatabase()</a></code> (see <a href="#accessing-the-contrast-database">Accessing the contrast
database</a>) .</p>
<p>The <code><a href="../reference/queryWithContrasts.html">queryWithContrasts()</a></code> function returns a list with
three elements per query contrast:</p>
<ul>
<li>“pearson.rhos” is itself a list with each element containing the
Pearson correlation values against all the <code>orthosData</code>
entries for a specific component (input, decoded (polytopic), residual
(specific)).</li>
<li>“zscores” is also a list with each element containing the z-score
transformed version of the “pearson.rhos” values.</li>
<li>“TopHits” is also a list with detailed <code>orthosData</code>
metadata for each of the top <code>detailTopn</code> hits per component
(default 10).</li>
</ul>
<p>In the following examples, please note that the queries are run using
<code>mode =  "DEMO"</code> in order to keep computations short. For
actual analyses, the default <code>mode = "ANALYSIS"</code> should be
used.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># parallelization parameters:</span></span>
<span><span class="va">params</span> <span class="op">&lt;-</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html" class="external-link">MulticoreParam</a></span><span class="op">(</span>workers <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># for demonstration purposes (for actual analyses, use 'mode = "ANALYSIS"'):</span></span>
<span><span class="va">query.res.human</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/queryWithContrasts.html">queryWithContrasts</a></span><span class="op">(</span><span class="va">dec_MKL1_human</span>, organism <span class="op">=</span> <span class="st">"Human"</span>, </span>
<span>                                      BPPARAM <span class="op">=</span> <span class="va">params</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                                      mode <span class="op">=</span> <span class="st">"DEMO"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">query.res.human</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">query.res.human</span><span class="op">$</span><span class="va">zscores</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># query contrasts in rows, `orthosData` entries in columns:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">query.res.human</span><span class="op">$</span><span class="va">zscores</span><span class="op">$</span><span class="va">RESIDUAL_CONTRASTS</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">query.res.human</span><span class="op">$</span><span class="va">zscores</span><span class="op">$</span><span class="va">RESIDUAL_CONTRASTS</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Information on the top hits of the query using the residual human MKL1/caMKL1 contrasts:</span></span>
<span><span class="va">query.res.human</span><span class="op">$</span><span class="va">TopHits</span><span class="op">$</span><span class="va">RESIDUAL_CONTRASTS</span></span>
<span></span>
<span><span class="co"># query the database using only the "caMKL1" mouse contrast, suppress plotting:</span></span>
<span><span class="co"># for demonstration purposes (for actual analyses, use 'mode = "ANALYSIS"'):</span></span>
<span><span class="va">query.res.mouse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/queryWithContrasts.html">queryWithContrasts</a></span><span class="op">(</span><span class="va">dec_MKL1_mouse</span><span class="op">[</span>, <span class="st">"caMKL1"</span><span class="op">]</span>, organism <span class="op">=</span> <span class="st">"Mouse"</span>, </span>
<span>                                      BPPARAM <span class="op">=</span> <span class="va">params</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                                      plotType <span class="op">=</span> <span class="st">"none"</span>, mode <span class="op">=</span> <span class="st">"DEMO"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot results for individual contrasts using violin plots:</span></span>
<span><span class="va">ViolinPlots_mouse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotQueryResultsViolin.html">plotQueryResultsViolin</a></span><span class="op">(</span><span class="va">query.res.mouse</span>, doPlot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">ViolinPlots_mouse</span><span class="op">[[</span><span class="st">"caMKL1"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># plot results for individual contrasts using composite Manhattan/Density plots:</span></span>
<span><span class="va">ManhDensPlots_mouse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotQueryResultsManh.html">plotQueryResultsManh</a></span><span class="op">(</span><span class="va">query.res.mouse</span>, doPlot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">ManhDensPlots_mouse</span><span class="op">[[</span><span class="st">"caMKL1"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span></span>
<span><span class="co">#Information on the top hits of the query using the residual mouse caMKL1 contrasts:</span></span>
<span><span class="va">query.res.mouse</span><span class="op">$</span><span class="va">TopHits</span><span class="op">$</span><span class="va">RESIDUAL_CONTRASTS</span></span></code></pre></div>
<p>The top hits obtained for the residual (specific) fractions of either
MKL1 or caMKL1 contrasts both in human and mouse are more clearly
separated from the background compared to those obtained from the input
or decoded (polytopic) fractions.</p>
<p>More importantly closer inspection of those top hits for the residual
contrasts in both experiments (e.g hits from series GSE77120, GSE112277
or GSE140898 in human or GSE164860 in mouse) reveal that they correspond
to treatments involving either MKL/MRTFA overexpression or
overexpression of the MKL related transcription factor MYOCD. Of note,
these treatments were performed in various cell contexts, different from
the ones of the MKL study under consideration (LM2 and 4T1 cell lines
for mouse and human respectively).</p>
<p>In general, as in this example, the residual sepcific fraction of a
DGE profile will be a better query proxy for molecularly and
mechanistically related treatments as it is largely stripped of nuisance
variance present in the original contrast.</p>
<p>On the other hand the decoded polytopic fraction and its
corresponding query hits can also be of interest in some applications as
it informs on downstream/secondary or lateral treatment effects.</p>
</div>
<div class="section level2">
<h2 id="accessing-the-contrast-database">Accessing the contrast database<a class="anchor" aria-label="anchor" href="#accessing-the-contrast-database"></a>
</h2>
<p>The <code>orthos</code> package provides functionality for direct
access to contrast databases of <code>orthosData</code> with the
<code><a href="../reference/loadContrastDatabase.html">loadContrastDatabase()</a></code> function.</p>
<p>This can be used to retrieve contrast values for all or subsets of
genes or metadata for specific datasets, e.g for hits identified with
<code><a href="../reference/queryWithContrasts.html">queryWithContrasts()</a></code>.</p>
<p>The organism-specific databases are compiled HDF5SummarizedExperiment
objects. As was the case for the <code>orthos</code> models, a database
will be automatically downloaded from <code>ExperimentHub</code> and
cached in the user ExperimentHub directory (see
<code>ExperimentHub::getExperimentHubOption("CACHE")</code>) using the
<code>orthosData</code> companion package, the first time
<code><a href="../reference/loadContrastDatabase.html">loadContrastDatabase()</a></code> is called for that database either
directly or via <code><a href="../reference/queryWithContrasts.html">queryWithContrasts()</a></code>.</p>
<p>The HDF5SummarizedExperiment object contains pre-calculated INPUT,
RESIDUAL and DECODED log2 fold change contrasts as well as the
corresponding expression CONTEXT in log2 CPM representation for all the
datasets in <code>orthosData</code>.</p>
<p>Extensive gene and contrast annotation is available in the object’s
<code>rowData</code> and <code>colData</code> respectively.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">organism</span> <span class="op">&lt;-</span> <span class="st">"Mouse"</span></span>
<span><span class="va">orthosDB</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/loadContrastDatabase.html">loadContrastDatabase</a></span> <span class="op">(</span>organism<span class="op">=</span><span class="st">"Mouse"</span>, mode<span class="op">=</span><span class="st">"DEMO"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">orthosDB</span> </span>
<span></span>
<span><span class="co">#Available contrast annotations:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">orthosDB</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Available gene annotations:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">orthosDB</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Retrieve partial annotation for a specific contrast</span></span>
<span><span class="co">#returned as a top-hit in the mouse caMKL1 query above:</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">orthosDB</span><span class="op">)</span><span class="op">[</span><span class="st">"GSM5021181"</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"title"</span>,<span class="st">"series_id"</span>,<span class="st">"CNTname"</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Compare context and individual contrast fractions between</span></span>
<span><span class="co"># the mouse caMKL1 contrast under consideration and the "GSM5021181"</span></span>
<span><span class="co"># query hit:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">queryID</span> <span class="op">&lt;-</span> <span class="st">"GSM5021181"</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">contrast</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">dec_MKL1_mouse</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">unit</span> <span class="op">&lt;-</span> <span class="st">"L2FC"</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">contrast</span><span class="op">==</span><span class="st">"CONTEXT"</span><span class="op">)</span> <span class="op">{</span><span class="va">unit</span> <span class="op">&lt;-</span> <span class="st">"L2CPM"</span><span class="op">}</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">dec_MKL1_mouse</span><span class="op">)</span><span class="op">[[</span><span class="va">contrast</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>, <span class="st">"caMKL1"</span><span class="op">]</span>,</span>
<span>         <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">orthosDB</span><span class="op">)</span><span class="op">[[</span><span class="va">contrast</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>, <span class="va">queryID</span><span class="op">]</span>,</span>
<span>         pch <span class="op">=</span> <span class="fl">16</span>, cex <span class="op">=</span> <span class="fl">0.5</span>, col <span class="op">=</span> <span class="st">"darkslategrey"</span>, main <span class="op">=</span> <span class="va">contrast</span>,</span>
<span>         xlab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">unit</span>, <span class="st">" caMKL1"</span><span class="op">)</span>, ylab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"LFC "</span>, <span class="va">queryID</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, col <span class="op">=</span> <span class="st">"darkred"</span>, lwd <span class="op">=</span> <span class="fl">0.8</span>, lty <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="op">}</span>     </span></code></pre></div>
</div>
<div class="section level2">
<h2 id="advanced-use-cases-accessing-the-model-internals">Advanced use cases: Accessing the model internals<a class="anchor" aria-label="anchor" href="#advanced-use-cases-accessing-the-model-internals"></a>
</h2>
<p>Producing latent encodings of contrasts Producing latent encodings of
contexts Morphing decoded contrasts according to a different context
Fine-tuning the model to custom contrast datasets</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS Monterey 12.6.5</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats    graphics utils    stats4   methods  base    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] orthos_0.99.0               SummarizedExperiment_1.30.0</span></span>
<span><span class="co">#&gt;  [3] Biobase_2.60.0              GenomicRanges_1.52.0       </span></span>
<span><span class="co">#&gt;  [5] GenomeInfoDb_1.36.0         IRanges_2.34.0             </span></span>
<span><span class="co">#&gt;  [7] S4Vectors_0.38.0            BiocGenerics_0.46.0        </span></span>
<span><span class="co">#&gt;  [9] MatrixGenerics_1.12.0       matrixStats_0.63.0         </span></span>
<span><span class="co">#&gt; [11] knitr_1.42                  BiocStyle_2.28.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] DBI_1.1.3                     bitops_1.0-7                 </span></span>
<span><span class="co">#&gt;   [3] rlang_1.1.0                   magrittr_2.0.3               </span></span>
<span><span class="co">#&gt;   [5] compiler_4.3.0                RSQLite_2.3.1                </span></span>
<span><span class="co">#&gt;   [7] dir.expiry_1.8.0              png_0.1-8                    </span></span>
<span><span class="co">#&gt;   [9] systemfonts_1.0.4             vctrs_0.6.2                  </span></span>
<span><span class="co">#&gt;  [11] stringr_1.5.0                 pkgconfig_2.0.3              </span></span>
<span><span class="co">#&gt;  [13] crayon_1.5.2                  fastmap_1.1.1                </span></span>
<span><span class="co">#&gt;  [15] backports_1.4.1               dbplyr_2.3.2                 </span></span>
<span><span class="co">#&gt;  [17] XVector_0.40.0                ellipsis_0.3.2               </span></span>
<span><span class="co">#&gt;  [19] fontawesome_0.5.1             utf8_1.2.3                   </span></span>
<span><span class="co">#&gt;  [21] promises_1.2.0.1              rmarkdown_2.21               </span></span>
<span><span class="co">#&gt;  [23] grDevices_4.3.0               ragg_1.2.5                   </span></span>
<span><span class="co">#&gt;  [25] purrr_1.0.1                   bit_4.0.5                    </span></span>
<span><span class="co">#&gt;  [27] xfun_0.39                     zlibbioc_1.46.0              </span></span>
<span><span class="co">#&gt;  [29] cachem_1.0.7                  jsonlite_1.8.4               </span></span>
<span><span class="co">#&gt;  [31] blob_1.2.4                    orthosData_0.1.0             </span></span>
<span><span class="co">#&gt;  [33] highr_0.10                    later_1.3.0                  </span></span>
<span><span class="co">#&gt;  [35] keras_2.11.1                  rhdf5filters_1.12.0          </span></span>
<span><span class="co">#&gt;  [37] DelayedArray_0.25.0           Rhdf5lib_1.22.0              </span></span>
<span><span class="co">#&gt;  [39] BiocParallel_1.34.0           interactiveDisplayBase_1.38.0</span></span>
<span><span class="co">#&gt;  [41] tensorflow_2.11.0             broom_1.0.4                  </span></span>
<span><span class="co">#&gt;  [43] parallel_4.3.0                R6_2.5.1                     </span></span>
<span><span class="co">#&gt;  [45] bslib_0.4.2                   stringi_1.7.12               </span></span>
<span><span class="co">#&gt;  [47] reticulate_1.28               car_3.1-2                    </span></span>
<span><span class="co">#&gt;  [49] jquerylib_0.1.4               Rcpp_1.0.10                  </span></span>
<span><span class="co">#&gt;  [51] bookdown_0.33                 base64enc_0.1-3              </span></span>
<span><span class="co">#&gt;  [53] httpuv_1.6.9                  Matrix_1.5-4                 </span></span>
<span><span class="co">#&gt;  [55] tidyselect_1.2.0              abind_1.4-5                  </span></span>
<span><span class="co">#&gt;  [57] yaml_2.3.7                    codetools_0.2-19             </span></span>
<span><span class="co">#&gt;  [59] curl_5.0.0                    plyr_1.8.8                   </span></span>
<span><span class="co">#&gt;  [61] lattice_0.21-8                tibble_3.2.1                 </span></span>
<span><span class="co">#&gt;  [63] shiny_1.7.4                   basilisk.utils_1.12.0        </span></span>
<span><span class="co">#&gt;  [65] KEGGREST_1.40.0               evaluate_0.20                </span></span>
<span><span class="co">#&gt;  [67] desc_1.4.2                    BiocFileCache_2.8.0          </span></span>
<span><span class="co">#&gt;  [69] ExperimentHub_2.8.0           Biostrings_2.68.0            </span></span>
<span><span class="co">#&gt;  [71] ggpubr_0.6.0                  pillar_1.9.0                 </span></span>
<span><span class="co">#&gt;  [73] BiocManager_1.30.20           filelock_1.0.2               </span></span>
<span><span class="co">#&gt;  [75] carData_3.0-5                 whisker_0.4.1                </span></span>
<span><span class="co">#&gt;  [77] generics_0.1.3                rprojroot_2.0.3              </span></span>
<span><span class="co">#&gt;  [79] RCurl_1.98-1.12               ggplot2_3.4.2                </span></span>
<span><span class="co">#&gt;  [81] BiocVersion_3.17.1            munsell_0.5.0                </span></span>
<span><span class="co">#&gt;  [83] scales_1.2.1                  xtable_1.8-4                 </span></span>
<span><span class="co">#&gt;  [85] glue_1.6.2                    tools_4.3.0                  </span></span>
<span><span class="co">#&gt;  [87] datasets_4.3.0                AnnotationHub_3.8.0          </span></span>
<span><span class="co">#&gt;  [89] ggsignif_0.6.4                fs_1.6.2                     </span></span>
<span><span class="co">#&gt;  [91] cowplot_1.1.1                 rhdf5_2.44.0                 </span></span>
<span><span class="co">#&gt;  [93] grid_4.3.0                    tidyr_1.3.0                  </span></span>
<span><span class="co">#&gt;  [95] AnnotationDbi_1.62.0          colorspace_2.1-0             </span></span>
<span><span class="co">#&gt;  [97] GenomeInfoDbData_1.2.10       basilisk_1.12.0              </span></span>
<span><span class="co">#&gt;  [99] HDF5Array_1.28.0              cli_3.6.1                    </span></span>
<span><span class="co">#&gt; [101] rappdirs_0.3.3                tfruns_1.5.1                 </span></span>
<span><span class="co">#&gt; [103] textshaping_0.3.6             fansi_1.0.4                  </span></span>
<span><span class="co">#&gt; [105] dplyr_1.1.2                   gtable_0.3.3                 </span></span>
<span><span class="co">#&gt; [107] ggsci_3.0.0                   rstatix_0.7.2                </span></span>
<span><span class="co">#&gt; [109] zeallot_0.1.0                 sass_0.4.5                   </span></span>
<span><span class="co">#&gt; [111] digest_0.6.31                 ggrepel_0.9.3                </span></span>
<span><span class="co">#&gt; [113] memoise_2.0.1                 htmltools_0.5.5              </span></span>
<span><span class="co">#&gt; [115] pkgdown_2.0.7                 lifecycle_1.0.3              </span></span>
<span><span class="co">#&gt; [117] httr_1.4.5                    mime_0.12                    </span></span>
<span><span class="co">#&gt; [119] bit64_4.0.5</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-hu2019mkl1" class="csl-entry">
Hu, Xiao, Zongzhi Z Liu, Xinyue Chen, Vincent P Schulz, Abhishek Kumar,
Amaleah A Hartman, Jason Weinstein, et al. 2019. <span>“MKL1-Actin
Pathway Restricts Chromatin Accessibility and Prevents Mature
Pluripotency Activation.”</span> <em>Nature Communications</em> 10 (1):
1695.
</div>
<div id="ref-lachmann2018massive" class="csl-entry">
Lachmann, Alexander, Denis Torre, Alexandra B Keenan, Kathleen M
Jagodnik, Hoyjin J Lee, Lily Wang, Moshe C Silverstein, and Avi Ma’ayan.
2018. <span>“Massive Mining of Publicly Available RNA-Seq Data from
Human and Mouse.”</span> <em>Nature Communications</em> 9 (1): 1366.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Panagiotis Papasaikas, Charlotte Soneson, Michael Stadler.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
