<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An introduction to contrast decomposition and querying using orthos • orthos</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An introduction to contrast decomposition and querying using orthos">
<meta property="og:description" content="orthos">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthos</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/orthosIntro.html">An introduction to contrast decomposition and querying using orthos</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>An introduction to contrast decomposition and
querying using orthos</h1>
                        <h4 data-toc-skip class="author">Panagiotis
Papasaikas</h4>
            <address class="author_afil">
      Friedrich Miescher Institute for Biomedical Research, Basel,
SwitzerlandSIB Swiss Institute of
Bioinformatics<br><a class="author_email" href="mailto:#"></a><a href="mailto:panagiotis.papasaikas@fmi.ch" class="email">panagiotis.papasaikas@fmi.ch</a>
      </address>
                              <h4 data-toc-skip class="author">Charlotte
Soneson</h4>
            <address class="author_afil">
      Friedrich Miescher Institute for Biomedical Research, Basel,
SwitzerlandSIB Swiss Institute of
Bioinformatics<br><a class="author_email" href="mailto:#"></a><a href="mailto:charlotte.soneson@fmi.ch" class="email">charlotte.soneson@fmi.ch</a>
      </address>
                              <h4 data-toc-skip class="author">Michael
Stadler</h4>
            <address class="author_afil">
      Friedrich Miescher Institute for Biomedical Research, Basel,
SwitzerlandSIB Swiss Institute of
Bioinformatics<br><a class="author_email" href="mailto:#"></a><a href="mailto:michael.stadler@fmi.ch" class="email">michael.stadler@fmi.ch</a>
      </address>
                  
            <h4 data-toc-skip class="date">31 May 2023</h4>
      
      
      <div class="hidden name"><code>orthosIntro.Rmd</code></div>

    </div>

    
    
<p><img src="orthos.png" style="width:50.0%"></p>
<style type="text/css">
body {
  margin: 0 auto;
  max-width: 1600px;
  padding: 2rem;
}
</style>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>RNAseq-based <strong>differential expression analysis upon cellular
perturbations</strong>, such as gene knockouts, RNA knockdowns or
compound treatment experiments, is the most commonly used tool for
probing molecular mechanisms of action due to its simplicity and low
cost.</p>
<p>However, interpretation of such gene expression contrasts is
confounded by the complex and nuanced impacts of experimental treatments
on cellular processes.</p>
<p>For example, knockout or over-expression of a transcription factor
will not only alter the transcription of its direct target genes, but
also cause many secondary expression changes. In addition, treatments or
treatment delivery agents typically elicit a variety of unintended,
systemic responses (such as immune, toxic, metabolic) that cannot be
well-controlled for by the design of the study.</p>
<p>The final experimentally measured gene expression changes are a hard
to assess convolution of <strong>specific</strong> and
<strong>non-specific</strong> secondary and lateral treatment
effects.</p>
<p><code>orthos</code> is a generative modelling-based approach that
disentangles the experiment-specific from the non-specific effects of
perturbations on gene expression. It is trained on a large corpus of
gene expression contrasts (per organism &gt;60K annotated, &gt;0.5M
augmented), compiled from the <a href="https://maayanlab.cloud/archs4/" class="external-link">ARCHS4</a> database of uniformly
processed RNAseq experiments (<span class="citation">Lachmann et al.
(2018)</span>). It accurately captures and isolates
<strong>non-specific</strong> effects (effects that are observed across
multiple treatments) while accounting for context (tissue or cell-line
experimental background).</p>
<p>The residual <strong>specific</strong> component obtained from this
decomposition offers a more unequivocal experimental signature and is
more closely related to the direct molecular effects of the perturbation
when compared to the raw signal.</p>
<p>In addition to providing a clearer understanding of the effects of
experimental treatments on gene expression, <code>orthos</code> also
enables researchers to <strong>query the contrast database</strong> with
arbitrary contrasts and identify experiments with similar specific
effects, ultimately helping to <strong>map treatments to mechanisms of
action</strong>.</p>
</div>
<div class="section level2">
<h2 id="installation-and-overview">Installation and overview<a class="anchor" aria-label="anchor" href="#installation-and-overview"></a>
</h2>
<p><code>orthos</code> can be installed from from Bioconductor using
<code><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">BiocManager::install()</a></code>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"orthos"</span><span class="op">)</span></span>
<span><span class="co"># or also...</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"orthos"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>After installation, the package can be loaded with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">orthos</span><span class="op">)</span></span></code></pre></div>
<p>A typical analysis involves two steps:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Decomposing</strong> one or several contrasts into their
corresponding specific and non-specific components using the
<code><a href="../reference/decomposeVar.html">decomposeVar()</a></code> function and</p></li>
<li><p><strong>Performing queries</strong> with the original and
decomposed specific and non-specific contrasts against the contrast
database using the <code><a href="../reference/queryWithContrasts.html">queryWithContrasts()</a></code> function.</p></li>
</ol>
</div>
<div class="section level2">
<h2 id="demonstration-data">Demonstration data<a class="anchor" aria-label="anchor" href="#demonstration-data"></a>
</h2>
<p>To demonstrate the functionality of <code>orthos</code> we use a
dataset from the the GEO series <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE215150" class="external-link">GSE215150</a>.</p>
<p>This series was not part of the <code>orthos</code> training or the
<code>orthosData</code> contrast database; it was only publicly released
on January 1st 2023 after the freeze of the training data to the ARCHS4
v2.1.2 database.</p>
<p>The performed experiment involves over-expression (OE) of the
MKL/megakaryoblastic leukemia 1 gene (also termed MRTFA/myocardin
related transcription factor A) and a constitutively active mutant MKL1
(caMKL1, described in <span class="citation">Hu et al. (2019)</span>).
Both OE experiments were performed in mouse LM2 and human 4T1
tumor-derived breast cancer cell lines. In addition to the MKL1/caMKL1
OE samples, the series also contains no-treatment controls for each of
the two cell lines.</p>
<p>For simplicity the three biological replicates of each available
condition have been collapsed in the data provided in the package.</p>
<p>In the provided form each of the two datasets (Mouse, Human) contains
raw counts for over 55,000 genes identified by gene symbols in three
conditions: Control (Ctrl), MKL1 OE (MKL1) and constitutively-active
MKL1 OE (caMKL1).</p>
<p>Load the human dataset:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MKL1_human</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GSE215150_MKL1_Human.rds"</span>,</span>
<span>                                  package <span class="op">=</span> <span class="st">"orthos"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">MKL1_human</span><span class="op">)</span></span>
<span><span class="co">#&gt;             Ctrl MKL1 caMKL1</span></span>
<span><span class="co">#&gt; DDX11L1        6    3      4</span></span>
<span><span class="co">#&gt; WASH7P        47   57     41</span></span>
<span><span class="co">#&gt; MIR6859-1     10   10      6</span></span>
<span><span class="co">#&gt; MIR1302-2HG    0    6      1</span></span>
<span><span class="co">#&gt; MIR1302-2      0    0      0</span></span>
<span><span class="co">#&gt; FAM138A        1    1      5</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">MKL1_human</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 59453     3</span></span></code></pre></div>
<p>Load the mouse dataset:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MKL1_mouse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GSE215150_MKL1_Mouse.rds"</span>,</span>
<span>                                  package <span class="op">=</span> <span class="st">"orthos"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">MKL1_mouse</span><span class="op">)</span></span>
<span><span class="co">#&gt;               Ctrl MKL1 caMKL1</span></span>
<span><span class="co">#&gt; 4933401J01Rik    0    0      0</span></span>
<span><span class="co">#&gt; Gm26206          0    0      0</span></span>
<span><span class="co">#&gt; Xkr4             0    0      0</span></span>
<span><span class="co">#&gt; Gm18956          0    0      0</span></span>
<span><span class="co">#&gt; Gm37180          0    0      0</span></span>
<span><span class="co">#&gt; Gm37363          0    0      0</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">MKL1_mouse</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 55367     3</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="decomposition-of-differential-gene-expression-variance-into-specific-and-non-specific-components-using-decomposevar">Decomposition of differential gene expression variance into specific
and non-specific components using <code>decomposeVar()</code><a class="anchor" aria-label="anchor" href="#decomposition-of-differential-gene-expression-variance-into-specific-and-non-specific-components-using-decomposevar"></a>
</h2>
<div class="section level3">
<h3 id="prelude-a-short-overview-of-the-orthos-models">Prelude: A short overview of the <code>orthos</code> models<a class="anchor" aria-label="anchor" href="#prelude-a-short-overview-of-the-orthos-models"></a>
</h3>
<p>The workhorse behind <code>orthos</code> are organism-specific
conditional variational autoencoder (cVAE) models that break down the
variance of a given differential expression experiment into a
non-specific and an experiment-specific component.</p>
<p>The non-specific component corresponds to gene variance that has been
observed across multiple instances during training, while the
experiment-specific variance is fairly unique to the experiment.</p>
<p>The inputs to the models are gene counts in the form of
log2-transformed counts per million (LCPMs) that are used to encode the
<strong>context</strong> of the performed experiment as well as the
actual gene expression <strong>contrasts</strong> in the form of log2
fold-changes (LFCs), i.e log2-transformed CPM ratios.</p>
<p>As we will see, calculation of those inputs is by default performed
internally given only <strong>raw gene counts</strong> and a
specification of the contrasted conditions.</p>
<p>Given these inputs, the model maps the contrast to a concise latent
representation (z<sub>D</sub>) which retains its recurring -and
therefore compressible- traits. The compressed latent representation is
then used to reconstruct a decoded version of the contrast<span style="color:blue">*</span>.</p>
<p>The <strong>decoded</strong> contrast corresponds directly to the
<strong>non-specific</strong> component of the observed effects; it
subsumes regularities i.e gene variance that the model can account for
because it has been repeatedly encountered, in some form, during
training.</p>
<p>The <strong>residual</strong> obtained after removing the decoded
contrast from the original one is the <strong>specific</strong>
component; this encompasses the gene variance that the model cannot
account for (experiment-specific biological effects + random noise).</p>
<p>From now on the terms decoded and non-specific will be used in
conjunction or interchangeably. Ditto for the terms residual and
specific.</p>
<p><img src="orthos_arch.png" style="width:100.0%" alt="orthos architecture"><span style="color:blue">*</span><font size='2"'><em> Notice that both the
latent encoding and the decoded output are conditioned on context (i.e
they are context-specific). This means that decomposing a contrast on a
different context will produce a different output. An interesting
ancillary application of this conditioning is that one can “morph”
decoded contrasts to <strong>in-silico evaluate non-specific effects in
new contexts</strong>.</em></font></p>
<p>In essence, we can infer what the non-specific effects would look
like had the experiment been performed in e.g different cell-lines,
tissues, batches, patients or using different library-preparation
protocols. The inferred decodings will reflect the new context in
multiple ways. For example, genes that were non-detected/non-expressed
in the original contrast (and therefore had neutral LFCs ) will produce
new (possibly non-neutral) decoded LFCs if present in the new context
and vice versa. This <strong>“out-of-context”</strong> type of inference
is limited to the decoded contrast, as by definition, the residual is
not part of the model’s generative capacity. Mechanically,
out-of-context inference of non-specific effects is a simple as
evaluating the same contrast using multiple contexts when calling
<code><a href="../reference/decomposeVar.html">decomposeVar()</a></code> (see section below). </p>
</div>
<div class="section level3">
<h3 id="contrast-decomposition-with-decomposevar">Contrast decomposition with <code>decomposeVar()</code><a class="anchor" aria-label="anchor" href="#contrast-decomposition-with-decomposevar"></a>
</h3>
<p>The function that performs this contrast decomposition into the
non-specific and specific components is <code><a href="../reference/decomposeVar.html">decomposeVar()</a></code>.
There are two available modes in which the required inputs can be fed
into the function:</p>
<ul>
<li><p>In the first mode the user passes the matrix <code>M</code> of
raw counts (genes in rows, conditions in columns) and two vectors
<code>treatm</code> and <code>cntr</code> of equal length specifying the
column indices in <code>M</code> corresponding to the “treatments” and
their respective “controls”. The same column indices can be repeated
multiple times in these vectors, for example in the case where multiple
treatments are paired to the same control.</p></li>
<li><p>In the second mode the user passes the matrix <code>M</code> of
raw counts and a second matrix <code>MD</code> that contains
pre-calculated log2 fold-changes for the contrasts to be analyzed. In
this mode, <code>M</code> specifies the contexts for the corresponding
columns of <code>MD</code>, and thus the two matrices need to have the
same dimensionality and identical row- and column-names. This would be
the mode of choice if e.g one wishes to produce the LFCs independently
or if one wants to evaluate the decoding of the same contrast(s) in
multiple contexts (e.g for “out-of-context” inference of non-specific
effects described above). In the latter case, copies of the same
contrast in columns of <code>MD</code> will be paired with columns of
<code>M</code> specifying the different contexts.</p></li>
</ul>
<p>In both modes the rownames of <code>M</code> (and <code>MD</code> if
specified) need to correspond to valid gene identifiers (currently
<code>orthos</code> supports Entrez gene identifiers, ENSEMBL gene
identifiers, gene symbols or ARCHS4 gene identifiers). By default the
type of gene identifier is detected automatically.</p>
<p>The first time that <code>decomposeVar</code> is executed for a
particular organism, the models required for inference will be
automatically downloaded from <code>ExperimentHub</code> and cached in
the user ExperimentHub directory (see
<code>ExperimentHub::getExperimentHubOption("CACHE")</code>) using the
<code>orthosData</code> companion package.</p>
<p>For the MKL1 data, which are stored as raw counts, it is more natural
to call <code><a href="../reference/decomposeVar.html">decomposeVar()</a></code> using the first mode.</p>
<p>Decomposing the human contrasts:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Decompose MKL1-vs-Cntrl and caMKL1-vs-Cntrl contrasts for human:</span></span>
<span><span class="va">dec_MKL1_human</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/decomposeVar.html">decomposeVar</a></span><span class="op">(</span>M <span class="op">=</span> <span class="va">MKL1_human</span>, treatm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span>, cntr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, </span>
<span>                               organism <span class="op">=</span> <span class="st">"Human"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">dec_MKL1_human</span></span>
<span><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 20411 2 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(4): INPUT_CONTRASTS DECODED_CONTRASTS RESIDUAL_CONTRASTS CONTEXT</span></span>
<span><span class="co">#&gt; rownames(20411): A1BG A1BG-AS1 ... WFDC5 XAGE2</span></span>
<span><span class="co">#&gt; rowData names(1): User_provided_IDs</span></span>
<span><span class="co">#&gt; colnames(2): MKL1 caMKL1</span></span>
<span><span class="co">#&gt; colData names(3): ACCOUNTED_VARIANCE.DECODED</span></span>
<span><span class="co">#&gt;   ACCOUNTED_VARIANCE.RESIDUAL ACCOUNTED_VARIANCE.COMMON</span></span></code></pre></div>
<p>Decomposing the mouse contrasts:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Decompose MKL1-vs-Cntrl and caMKL1-vs-Cntrl contrasts for mouse:</span></span>
<span><span class="va">dec_MKL1_mouse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/decomposeVar.html">decomposeVar</a></span><span class="op">(</span>M <span class="op">=</span> <span class="va">MKL1_mouse</span>, treatm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span>, cntr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>                               organism <span class="op">=</span> <span class="st">"Mouse"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">dec_MKL1_mouse</span></span>
<span><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 20339 2 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(4): INPUT_CONTRASTS DECODED_CONTRASTS RESIDUAL_CONTRASTS CONTEXT</span></span>
<span><span class="co">#&gt; rownames(20339): 0610005C13Rik 0610009B22Rik ... n-R5s146 n-R5s149</span></span>
<span><span class="co">#&gt; rowData names(1): User_provided_IDs</span></span>
<span><span class="co">#&gt; colnames(2): MKL1 caMKL1</span></span>
<span><span class="co">#&gt; colData names(3): ACCOUNTED_VARIANCE.DECODED</span></span>
<span><span class="co">#&gt;   ACCOUNTED_VARIANCE.RESIDUAL ACCOUNTED_VARIANCE.COMMON</span></span></code></pre></div>
<p>The output of <code><a href="../reference/decomposeVar.html">decomposeVar()</a></code> is a
<code>SummarizedExperiment</code> object with dimensions <code>N</code>
x <code>M</code>, where <code>N</code> is the number of
<code>orthos</code> genes<span style="color:blue">*</span> for that
organism and <code>M</code> is the number of contrasts specified during
input.</p>
<p>The <code>SummarizedExperiment</code> output also has 4 assay slots
corresponding to the input contrasts, decoded (non-specific), and
residual (specific) components, as well as the gene context. Contrasts
are represented as log2 fold-changes (LFCs) and context is represented
as log2-transformed counts per million (log2 CPM).</p>
<p>We can use the returned object to produce an MA plot for the original
contrast or to check how the input and decomposed contrasts are related
to each other.</p>
<p>For example for the mouse caMKL1 contrast:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">dec_MKL1_mouse</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of length 4</span></span>
<span><span class="co">#&gt; names(4): INPUT_CONTRASTS DECODED_CONTRASTS RESIDUAL_CONTRASTS CONTEXT</span></span>
<span></span>
<span><span class="co">#MA plot of for the input contrasts:</span></span>
<span><span class="va">DF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>L2CPM<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">dec_MKL1_mouse</span>,<span class="st">"CONTEXT"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>,</span>
<span>                 L2FC_INPUT<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">dec_MKL1_mouse</span>,<span class="st">"INPUT_CONTRASTS"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>,</span>
<span>                 L2FC_DECODED<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">dec_MKL1_mouse</span>,<span class="st">"DECODED_CONTRASTS"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>,</span>
<span>                 L2FC_RESIDUAL<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">dec_MKL1_mouse</span>,<span class="st">"RESIDUAL_CONTRASTS"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span></span>
<span>                 <span class="op">)</span></span>
<span></span>
<span><span class="co">#MA plot of for the input contrast</span></span>
<span><span class="va">P1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">DF</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">L2CPM</span>, y<span class="op">=</span><span class="va">L2FC_INPUT</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.4</span>, size<span class="op">=</span><span class="fl">1.8</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"darkgray"</span>, linewidth <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Expression (Log2 CPMs)"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Log2 Fold Change"</span><span class="op">)</span>  </span>
<span></span>
<span><span class="co">#Delta-delta plots for the input and decomposed contrast fractions</span></span>
<span><span class="va">P2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">DF</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">L2FC_INPUT</span>, y<span class="op">=</span><span class="va">L2FC_DECODED</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.4</span>, size<span class="op">=</span><span class="fl">1.8</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"darkgray"</span>, linewidth <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Log2 Fold Change INPUT"</span><span class="op">)</span>  <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Log2 Fold Change DECODED"</span><span class="op">)</span>  </span>
<span></span>
<span><span class="va">P3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">DF</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">L2FC_INPUT</span>, y<span class="op">=</span><span class="va">L2FC_RESIDUAL</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.4</span>, size<span class="op">=</span><span class="fl">1.8</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"darkgray"</span>, linewidth <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Log2 Fold Change INPUT"</span><span class="op">)</span>  <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Log2 Fold Change RESIDUAL"</span><span class="op">)</span>  </span>
<span></span>
<span><span class="va">P4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">DF</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">L2FC_DECODED</span>, y<span class="op">=</span><span class="va">L2FC_RESIDUAL</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha<span class="op">=</span><span class="fl">0.4</span>, size<span class="op">=</span><span class="fl">1.8</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"darkgray"</span>, linewidth <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Log2 Fold Change DECODED"</span><span class="op">)</span>  <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Log2 Fold Change RESIDUAL"</span><span class="op">)</span>  </span>
<span></span>
<span><span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">P1</span>,<span class="va">P2</span>,<span class="va">P3</span>,<span class="va">P4</span><span class="op">)</span></span></code></pre></div>
<p><img src="orthosIntro_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>As expected, both the decoded and residual components are correlated
to the input contrast. However, the residual and decoded components are
largely uncorrelated.</p>
<p>The <code>colData</code> of the object summarizes the proportion of
variance accounted for in each decomposed component:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">dec_MKL1_human</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 2 rows and 3 columns</span></span>
<span><span class="co">#&gt;        ACCOUNTED_VARIANCE.DECODED ACCOUNTED_VARIANCE.RESIDUAL</span></span>
<span><span class="co">#&gt;                         &lt;numeric&gt;                   &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; MKL1                     0.326622                    0.834041</span></span>
<span><span class="co">#&gt; caMKL1                   0.323083                    0.776842</span></span>
<span><span class="co">#&gt;        ACCOUNTED_VARIANCE.COMMON</span></span>
<span><span class="co">#&gt;                        &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; MKL1                   0.1606626</span></span>
<span><span class="co">#&gt; caMKL1                 0.0999251</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">dec_MKL1_mouse</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 2 rows and 3 columns</span></span>
<span><span class="co">#&gt;        ACCOUNTED_VARIANCE.DECODED ACCOUNTED_VARIANCE.RESIDUAL</span></span>
<span><span class="co">#&gt;                         &lt;numeric&gt;                   &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; MKL1                     0.340694                    0.873229</span></span>
<span><span class="co">#&gt; caMKL1                   0.294274                    0.791121</span></span>
<span><span class="co">#&gt;        ACCOUNTED_VARIANCE.COMMON</span></span>
<span><span class="co">#&gt;                        &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; MKL1                   0.2139228</span></span>
<span><span class="co">#&gt; caMKL1                 0.0853951</span></span></code></pre></div>
<p><span style="color:blue">*</span><font size='2"'><em> Notice that, of
the total gene features present in the input (over 55,000), only ~20,000
genes are part of the <code>orthos</code> model () and the
<code><a href="../reference/decomposeVar.html">decomposeVar()</a></code> output.</em></font></p>
<p>These ~20,000 <code>orthos</code> genes are “sanctioned” according to
several criteria (located on canonical chromosomes, no pseudogenes, no
ribosomal protein genes, detected in at least a small fraction of the
ARCHS4 database).</p>
<p>The model is highly robust to small fractions of <code>orthos</code>
genes not being part of the user input, even if those genes are
expressed in the context under consideration. That being noted, it is
safer to feed-in inputs that are as comprehensive as possible, i.e
<strong>not filtered in any way</strong>, in terms of gene features.
</p>
</div>
</div>
<div class="section level2">
<h2 id="querying-the-database-of-gene-contrasts-using-querywithcontrasts">Querying the database of gene contrasts using
<code>queryWithContrasts()</code><a class="anchor" aria-label="anchor" href="#querying-the-database-of-gene-contrasts-using-querywithcontrasts"></a>
</h2>
<p>Typically, the next step of the analysis involves querying the
contrasts database (<code>orthosData</code>) to identify public
experiments similar to the one(s) under investigation, either in terms
of the original or decomposed decoded (non-specific) and residual
(specific) contrasts. As we will see in the following examples the
results’ of these queries can guide the interpretation of of the
different contrast fractions.</p>
<p><code>orthosData</code> contains over 100,000 differential gene
expression experiments compiled from the ARCHS4 database of publicly
available expression data (<span class="citation">Lachmann et al.
(2018)</span>). Each entry in <code>orthosData</code> corresponds to a
pair of RNAseq samples contrasting a treatment vs a control condition. A
combination of metadata, semantic and quantitative analyses was used to
determine the proper assignment of samples to such pairs in
<code>orthosData</code>.</p>
<p>The function that performs the queries against
<code>orthosData</code> is <code><a href="../reference/queryWithContrasts.html">queryWithContrasts()</a></code>. The input
to this function is the <code>SummarizedExperiment</code> object
obtained in the previous step from <code><a href="../reference/decomposeVar.html">decomposeVar()</a></code>, either
the complete object or one that has been column-subsetted, allowing to
query the contrast database with only a subset of the decomposed
contrasts.</p>
<p>As was the case for the <code>orthos</code> models, a database will
be automatically downloaded from <code>ExperimentHub</code> and cached
in the user ExperimentHub directory (see
<code>ExperimentHub::getExperimentHubOption("CACHE")</code>) using the
<code>orthosData</code> companion package, the first time
<code><a href="../reference/queryWithContrasts.html">queryWithContrasts()</a></code> is called for that database or the
first time the user attempts to access the database directly with
<code><a href="../reference/loadContrastDatabase.html">loadContrastDatabase()</a></code> (see <a href="#accessing-the-contrast-database">Accessing the contrast
database</a>) .</p>
<p>The <code><a href="../reference/queryWithContrasts.html">queryWithContrasts()</a></code> function returns a list with
three elements per query contrast:</p>
<ul>
<li>“pearson.rhos” is itself a list with each element containing the
Pearson correlation values against all the <code>orthosData</code>
entries for a specific component (input, decoded/non-specific,
residual/specific).</li>
<li>“zscores” is also a list with each element containing the z-score
transformed version of the “pearson.rhos” values.</li>
<li>“TopHits” is also a list with detailed <code>orthosData</code>
metadata for each of the top <code>detailTopn</code> hits per component
(default 10).</li>
</ul>
<p>In the following examples, please note that the queries are run using
<code>mode =  "DEMO"</code> in order to keep computations short. For
actual analyses, the default <code>mode = "ANALYSIS"</code> should be
used.</p>
<p>Examples queries using the decomposed human MKL1 data:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># parallelization parameters:</span></span>
<span><span class="va">params</span> <span class="op">&lt;-</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html" class="external-link">MulticoreParam</a></span><span class="op">(</span>workers <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># for demonstration purposes (for actual analyses, use 'mode = "ANALYSIS"'):</span></span>
<span><span class="va">query.res.human</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/queryWithContrasts.html">queryWithContrasts</a></span><span class="op">(</span><span class="va">dec_MKL1_human</span>, organism <span class="op">=</span> <span class="st">"Human"</span>, </span>
<span>                                      BPPARAM <span class="op">=</span> <span class="va">params</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                                      mode <span class="op">=</span> <span class="st">"DEMO"</span><span class="op">)</span></span></code></pre></div>
<p><img src="orthosIntro_files/figure-html/unnamed-chunk-8-1.png" width="1152"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">query.res.human</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "pearson.rhos" "zscores"      "TopHits"</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">query.res.human</span><span class="op">$</span><span class="va">zscores</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "INPUT_CONTRASTS"    "DECODED_CONTRASTS"  "RESIDUAL_CONTRASTS"</span></span>
<span></span>
<span><span class="co"># query contrasts in rows, `orthosData` entries in columns:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">query.res.human</span><span class="op">$</span><span class="va">zscores</span><span class="op">$</span><span class="va">RESIDUAL_CONTRASTS</span><span class="op">)</span> </span>
<span><span class="co">#&gt; [1]    2 1000</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">query.res.human</span><span class="op">$</span><span class="va">zscores</span><span class="op">$</span><span class="va">RESIDUAL_CONTRASTS</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;       MKL1              caMKL1        </span></span>
<span><span class="co">#&gt;  Min.   :-3.17999   Min.   :-2.47869  </span></span>
<span><span class="co">#&gt;  1st Qu.:-0.53837   1st Qu.:-0.44872  </span></span>
<span><span class="co">#&gt;  Median :-0.09769   Median :-0.09039  </span></span>
<span><span class="co">#&gt;  Mean   : 0.00000   Mean   : 0.00000  </span></span>
<span><span class="co">#&gt;  3rd Qu.: 0.36729   3rd Qu.: 0.30576  </span></span>
<span><span class="co">#&gt;  Max.   : 7.38595   Max.   :13.90709</span></span>
<span></span>
<span><span class="co">#Information on the top hits of the query using the residual human MKL1/caMKL1 contrasts:</span></span>
<span><span class="va">query.res.human</span><span class="op">$</span><span class="va">TopHits</span><span class="op">$</span><span class="va">RESIDUAL_CONTRASTS</span></span>
<span><span class="co">#&gt; $MKL1</span></span>
<span><span class="co">#&gt; DataFrame with 10 rows and 8 columns</span></span>
<span><span class="co">#&gt;               Zscore TREATM_geo_accession                  title</span></span>
<span><span class="co">#&gt;            &lt;numeric&gt;          &lt;character&gt;            &lt;character&gt;</span></span>
<span><span class="co">#&gt; GSM2044531   7.38595           GSM2044531                MYOCD-1</span></span>
<span><span class="co">#&gt; GSM2044532   7.31258           GSM2044532                MYOCD-2</span></span>
<span><span class="co">#&gt; GSM3066023   6.63946           GSM3066023 HEL-iMKL1 Dox treate..</span></span>
<span><span class="co">#&gt; GSM3544074   6.21459           GSM3544074        MEL270.SF3B1_WT</span></span>
<span><span class="co">#&gt; GSM4869874   5.46485           GSM4869874 U2OS cells - DMSO - ..</span></span>
<span><span class="co">#&gt; GSM3544069   5.23304           GSM3544069     MEL270.SF3B1_K700E</span></span>
<span><span class="co">#&gt; GSM5013969   4.91198           GSM5013969 chase_4sU_merged.EtO..</span></span>
<span><span class="co">#&gt; GSM4718649   4.79887           GSM4718649 F508delCFBE41o, siCT..</span></span>
<span><span class="co">#&gt; GSM5013974   4.79434           GSM5013974 nochase_spiked_4sU_4..</span></span>
<span><span class="co">#&gt; GSM5709388   4.73799           GSM5709388   EFM192A-LAP-DTP1_S11</span></span>
<span><span class="co">#&gt;               characteristics_ch1           series_id        source_name_ch1</span></span>
<span><span class="co">#&gt;                       &lt;character&gt;         &lt;character&gt;            &lt;character&gt;</span></span>
<span><span class="co">#&gt; GSM2044531 transduction: ad-MYOCD            GSE77120 vascular smooth musc..</span></span>
<span><span class="co">#&gt; GSM2044532 transduction: ad-MYOCD            GSE77120 vascular smooth musc..</span></span>
<span><span class="co">#&gt; GSM3066023 treatment: Treated w.. GSE112277,GSE112279        HEL-iMKL1 cells</span></span>
<span><span class="co">#&gt; GSM3544074 cell line: MEL270,le..           GSE124720                 MEL270</span></span>
<span><span class="co">#&gt; GSM4869874 cell type: U2OS,trea.. GSE160253,GSE160254             U2OS cells</span></span>
<span><span class="co">#&gt; GSM3544069 cell line: MEL270,le..           GSE124720                 MEL270</span></span>
<span><span class="co">#&gt; GSM5013969 treatment (etoh or 4.. GSE164555,GSE164569 SH-EP-MYCNER shEXOSC10</span></span>
<span><span class="co">#&gt; GSM4718649 cell type: Immortali..           GSE155987 Parental CFBE41o- Ce..</span></span>
<span><span class="co">#&gt; GSM5013974 treatment (etoh or 4.. GSE164555,GSE164569 SH-EP-MYCNER shEXOSC10</span></span>
<span><span class="co">#&gt; GSM5709388 treatment: 14 day la..           GSE156246                EFM192A</span></span>
<span><span class="co">#&gt;            CNT_geo_accession corr_TREATM_CNT</span></span>
<span><span class="co">#&gt;                  &lt;character&gt;       &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; GSM2044531        GSM2044530        0.963492</span></span>
<span><span class="co">#&gt; GSM2044532        GSM2044530        0.963938</span></span>
<span><span class="co">#&gt; GSM3066023        GSM3066021        0.995455</span></span>
<span><span class="co">#&gt; GSM3544074        GSM3544065        0.995561</span></span>
<span><span class="co">#&gt; GSM4869874        GSM4869870        0.993011</span></span>
<span><span class="co">#&gt; GSM3544069        GSM3544065        0.994787</span></span>
<span><span class="co">#&gt; GSM5013969        GSM5013971        0.994540</span></span>
<span><span class="co">#&gt; GSM4718649        GSM4718652        0.993717</span></span>
<span><span class="co">#&gt; GSM5013974        GSM5013976        0.992348</span></span>
<span><span class="co">#&gt; GSM5709388        GSM5709390        0.959956</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $caMKL1</span></span>
<span><span class="co">#&gt; DataFrame with 10 rows and 8 columns</span></span>
<span><span class="co">#&gt;               Zscore TREATM_geo_accession                  title</span></span>
<span><span class="co">#&gt;            &lt;numeric&gt;          &lt;character&gt;            &lt;character&gt;</span></span>
<span><span class="co">#&gt; GSM2044531  13.90709           GSM2044531                MYOCD-1</span></span>
<span><span class="co">#&gt; GSM2044532  13.85102           GSM2044532                MYOCD-2</span></span>
<span><span class="co">#&gt; GSM4189923   6.84975           GSM4189923      HT11_iSM_old_rep3</span></span>
<span><span class="co">#&gt; GSM4189922   6.83691           GSM4189922      HT15_iSM_old_rep2</span></span>
<span><span class="co">#&gt; GSM3976594   4.62126           GSM3976594         FSK_IL-1beta_2</span></span>
<span><span class="co">#&gt; GSM3066023   4.62013           GSM3066023 HEL-iMKL1 Dox treate..</span></span>
<span><span class="co">#&gt; GSM3976593   4.52470           GSM3976593         FSK_IL-1beta_1</span></span>
<span><span class="co">#&gt; GSM3976595   4.38356           GSM3976595         FSK_IL-1beta_3</span></span>
<span><span class="co">#&gt; GSM4189918   3.97301           GSM4189918    RS17_iSM_young_rep1</span></span>
<span><span class="co">#&gt; GSM4189920   3.89561           GSM4189920    RS14_iSM_young_rep3</span></span>
<span><span class="co">#&gt;               characteristics_ch1           series_id        source_name_ch1</span></span>
<span><span class="co">#&gt;                       &lt;character&gt;         &lt;character&gt;            &lt;character&gt;</span></span>
<span><span class="co">#&gt; GSM2044531 transduction: ad-MYOCD            GSE77120 vascular smooth musc..</span></span>
<span><span class="co">#&gt; GSM2044532 transduction: ad-MYOCD            GSE77120 vascular smooth musc..</span></span>
<span><span class="co">#&gt; GSM4189923 treatment: induced c..           GSE140898 human induced smooth..</span></span>
<span><span class="co">#&gt; GSM4189922 treatment: induced c..           GSE140898 human induced smooth..</span></span>
<span><span class="co">#&gt; GSM3976594 cell type: hTERT-HM^..           GSE134896             myometrium</span></span>
<span><span class="co">#&gt; GSM3066023 treatment: Treated w.. GSE112277,GSE112279        HEL-iMKL1 cells</span></span>
<span><span class="co">#&gt; GSM3976593 cell type: hTERT-HM^..           GSE134896             myometrium</span></span>
<span><span class="co">#&gt; GSM3976595 cell type: hTERT-HM^..           GSE134896             myometrium</span></span>
<span><span class="co">#&gt; GSM4189918 treatment: induced c..           GSE140898 human induced smooth..</span></span>
<span><span class="co">#&gt; GSM4189920 treatment: induced c..           GSE140898 human induced smooth..</span></span>
<span><span class="co">#&gt;            CNT_geo_accession corr_TREATM_CNT</span></span>
<span><span class="co">#&gt;                  &lt;character&gt;       &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; GSM2044531        GSM2044530        0.963492</span></span>
<span><span class="co">#&gt; GSM2044532        GSM2044530        0.963938</span></span>
<span><span class="co">#&gt; GSM4189923        GSM4189934        0.975019</span></span>
<span><span class="co">#&gt; GSM4189922        GSM4189934        0.978935</span></span>
<span><span class="co">#&gt; GSM3976594        GSM3976579        0.985832</span></span>
<span><span class="co">#&gt; GSM3066023        GSM3066021        0.995455</span></span>
<span><span class="co">#&gt; GSM3976593        GSM3976579        0.987186</span></span>
<span><span class="co">#&gt; GSM3976595        GSM3976579        0.987181</span></span>
<span><span class="co">#&gt; GSM4189918        GSM4189933        0.983446</span></span>
<span><span class="co">#&gt; GSM4189920        GSM4189934        0.979353</span></span></code></pre></div>
<p>Example queries using the decomposed mouse MKL1 data:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># query the database using only the "caMKL1" mouse contrast, suppress plotting:</span></span>
<span><span class="co"># for demonstration purposes (for actual analyses, use 'mode = "ANALYSIS"'):</span></span>
<span><span class="va">query.res.mouse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/queryWithContrasts.html">queryWithContrasts</a></span><span class="op">(</span><span class="va">dec_MKL1_mouse</span><span class="op">[</span>, <span class="st">"caMKL1"</span><span class="op">]</span>, organism <span class="op">=</span> <span class="st">"Mouse"</span>, </span>
<span>                                      BPPARAM <span class="op">=</span> <span class="va">params</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                                      plotType <span class="op">=</span> <span class="st">"none"</span>, mode <span class="op">=</span> <span class="st">"DEMO"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot results for individual contrasts using violin plots:</span></span>
<span><span class="va">ViolinPlots_mouse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotQueryResultsViolin.html">plotQueryResultsViolin</a></span><span class="op">(</span><span class="va">query.res.mouse</span>, doPlot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">ViolinPlots_mouse</span><span class="op">[[</span><span class="st">"caMKL1"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p><img src="orthosIntro_files/figure-html/unnamed-chunk-9-1.png" width="1152"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># plot results for individual contrasts using composite Manhattan/Density plots:</span></span>
<span><span class="va">ManhDensPlots_mouse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotQueryResultsManh.html">plotQueryResultsManh</a></span><span class="op">(</span><span class="va">query.res.mouse</span>, doPlot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">ManhDensPlots_mouse</span><span class="op">[[</span><span class="st">"caMKL1"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p><img src="orthosIntro_files/figure-html/unnamed-chunk-9-2.png" width="1152"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span></span>
<span><span class="co">#Information on the top hits of the query using the residual mouse caMKL1 contrasts:</span></span>
<span><span class="va">query.res.mouse</span><span class="op">$</span><span class="va">TopHits</span><span class="op">$</span><span class="va">RESIDUAL_CONTRASTS</span></span>
<span><span class="co">#&gt; $caMKL1</span></span>
<span><span class="co">#&gt; DataFrame with 10 rows and 8 columns</span></span>
<span><span class="co">#&gt;               Zscore TREATM_geo_accession                  title</span></span>
<span><span class="co">#&gt;            &lt;numeric&gt;          &lt;character&gt;            &lt;character&gt;</span></span>
<span><span class="co">#&gt; GSM5021181  14.91050           GSM5021181 E0771_pRetrox_MRTFA_..</span></span>
<span><span class="co">#&gt; GSM5021173   9.44599           GSM5021173 B16F10_pRetrox_MRTFA..</span></span>
<span><span class="co">#&gt; GSM5021171   7.65648           GSM5021171 B16F10_pRetrox_MRTFA..</span></span>
<span><span class="co">#&gt; GSM5021172   5.90237           GSM5021172 B16F10_pRetrox_MRTFA..</span></span>
<span><span class="co">#&gt; GSM5396603   5.09239           GSM5396603                    NA4</span></span>
<span><span class="co">#&gt; GSM5396607   5.05168           GSM5396607                    NA8</span></span>
<span><span class="co">#&gt; GSM5396602   5.01185           GSM5396602                    NA3</span></span>
<span><span class="co">#&gt; GSM5396606   4.93570           GSM5396606                    NA7</span></span>
<span><span class="co">#&gt; GSM4928843   4.68661           GSM4928843 Î²-actin knockout Mo..</span></span>
<span><span class="co">#&gt; GSM3176949   4.61862           GSM3176949    11D_mmu-miR-1199-5p</span></span>
<span><span class="co">#&gt;               characteristics_ch1           series_id        source_name_ch1</span></span>
<span><span class="co">#&gt;                       &lt;character&gt;         &lt;character&gt;            &lt;character&gt;</span></span>
<span><span class="co">#&gt; GSM5021181 strain: C57BL6,cell ..           GSE164860            E0771 cells</span></span>
<span><span class="co">#&gt; GSM5021173 strain: C57BL6,cell ..           GSE164860           B16F10 cells</span></span>
<span><span class="co">#&gt; GSM5021171 strain: C57BL6,cell ..           GSE164860           B16F10 cells</span></span>
<span><span class="co">#&gt; GSM5021172 strain: C57BL6,cell ..           GSE164860           B16F10 cells</span></span>
<span><span class="co">#&gt; GSM5396603 strain: C57BL/6,tiss..           GSE178725 Mouse embryonic fibr..</span></span>
<span><span class="co">#&gt; GSM5396607 strain: C57BL/6,tiss..           GSE178725 Mouse embryonic fibr..</span></span>
<span><span class="co">#&gt; GSM5396602 strain: C57BL/6,tiss..           GSE178725 Mouse embryonic fibr..</span></span>
<span><span class="co">#&gt; GSM5396606 strain: C57BL/6,tiss..           GSE178725 Mouse embryonic fibr..</span></span>
<span><span class="co">#&gt; GSM4928843 cell type: fibroblas..           GSE161964 Mouse Embronic Fibro..</span></span>
<span><span class="co">#&gt; GSM3176949 cell line: NMUMG,sub.. GSE115375,GSE115376  NMUMG cells, clone E9</span></span>
<span><span class="co">#&gt;            CNT_geo_accession corr_TREATM_CNT</span></span>
<span><span class="co">#&gt;                  &lt;character&gt;       &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; GSM5021181        GSM5021179        0.974858</span></span>
<span><span class="co">#&gt; GSM5021173        GSM5021170        0.995724</span></span>
<span><span class="co">#&gt; GSM5021171        GSM5021168        0.994685</span></span>
<span><span class="co">#&gt; GSM5021172        GSM5021169        0.994397</span></span>
<span><span class="co">#&gt; GSM5396603        GSM5396601        0.993751</span></span>
<span><span class="co">#&gt; GSM5396607        GSM5396601        0.992054</span></span>
<span><span class="co">#&gt; GSM5396602        GSM5396600        0.993440</span></span>
<span><span class="co">#&gt; GSM5396606        GSM5396600        0.991412</span></span>
<span><span class="co">#&gt; GSM4928843        GSM4928841        0.975952</span></span>
<span><span class="co">#&gt; GSM3176949        GSM3176958        0.977919</span></span></code></pre></div>
<p>The top hits obtained for the residual (specific) fractions of either
MKL1 or caMKL1 contrasts both in human and mouse are more clearly
separated from the background compared to those obtained from the input
or decoded (non-specific) fractions.</p>
<p>More importantly closer inspection of those top hits for the residual
contrasts in both experiments (e.g hits from series GSE77120, GSE112277
or GSE140898 in human or GSE164860 in mouse) reveal that they correspond
to treatments involving either MKL/MRTFA overexpression or
overexpression of the MKL related transcription factor MYOCD. Of note,
these treatments were performed in various cell contexts, different from
the ones of the MKL study under consideration (LM2 and 4T1 cell lines
for mouse and human respectively).</p>
<p>In general, as in this example, the residual specific fraction of a
DGE profile will be a better query proxy for <strong>molecularly and
mechanistically related</strong> treatments as it is largely stripped of
nuisance variance present in the original contrast.</p>
<p>On the other hand the decoded non-specific fraction and its
corresponding query hits can also be of interest in some applications as
they provide information on the extent and type of
<strong>downstream/secondary or lateral</strong> treatment effects.</p>
</div>
<div class="section level2">
<h2 id="accessing-the-contrast-database">Accessing the contrast database<a class="anchor" aria-label="anchor" href="#accessing-the-contrast-database"></a>
</h2>
<p>The <code>orthos</code> package provides functionality for direct
access to contrast databases of <code>orthosData</code> with the
<code><a href="../reference/loadContrastDatabase.html">loadContrastDatabase()</a></code> function.</p>
<p>This can be used to retrieve contrast values for all or subsets of
genes or metadata for specific datasets, e.g for hits identified with
<code><a href="../reference/queryWithContrasts.html">queryWithContrasts()</a></code>.</p>
<p>The organism-specific databases are compiled as
HDF5SummarizedExperiment objects. As was the case for the
<code>orthos</code> models, a database will be automatically downloaded
from <code>ExperimentHub</code> and cached in the user ExperimentHub
directory (see
<code>ExperimentHub::getExperimentHubOption("CACHE")</code>) using the
<code>orthosData</code> companion package, the first time
<code><a href="../reference/loadContrastDatabase.html">loadContrastDatabase()</a></code> is called for that database either
directly or via <code><a href="../reference/queryWithContrasts.html">queryWithContrasts()</a></code>.</p>
<p>The HDF5SummarizedExperiment object contains pre-calculated INPUT,
RESIDUAL and DECODED log2 fold change contrasts as well as the
corresponding expression CONTEXT in log2 CPM representation for all the
datasets in <code>orthosData</code>.</p>
<p>Extensive gene and contrast annotation is available in the object’s
<code>rowData</code> and <code>colData</code> respectively.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">organism</span> <span class="op">&lt;-</span> <span class="st">"Mouse"</span></span>
<span><span class="va">orthosDB</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/loadContrastDatabase.html">loadContrastDatabase</a></span><span class="op">(</span>organism <span class="op">=</span> <span class="st">"Mouse"</span>, mode <span class="op">=</span> <span class="st">"DEMO"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">orthosDB</span> </span>
<span><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 20339 988 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(4): INPUT_CONTRASTS DECODED_CONTRASTS RESIDUAL_CONTRASTS CONTEXT</span></span>
<span><span class="co">#&gt; rownames(20339): 0610005C13Rik 0610009B22Rik ... n-R5s146 n-R5s149</span></span>
<span><span class="co">#&gt; rowData names(15): seqnames start ... ENTREZ_GENE_ID ARCHS4_ID</span></span>
<span><span class="co">#&gt; colnames(988): GSM1053282 GSM1061179 ... GSM970471 GSM984548</span></span>
<span><span class="co">#&gt; colData names(33): aligned_reads channel_count ... HasAssignedCNT</span></span>
<span><span class="co">#&gt;   CNTname</span></span>
<span></span>
<span><span class="co">#Available contrast annotations:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">orthosDB</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "aligned_reads"         "channel_count"         "characteristics_ch1"  </span></span>
<span><span class="co">#&gt;  [4] "contact_address"       "contact_city"          "contact_country"      </span></span>
<span><span class="co">#&gt;  [7] "contact_institute"     "contact_name"          "contact_zip"          </span></span>
<span><span class="co">#&gt; [10] "data_processing"       "extract_protocol_ch1"  "geo_accession"        </span></span>
<span><span class="co">#&gt; [13] "instrument_model"      "last_update_date"      "library_selection"    </span></span>
<span><span class="co">#&gt; [16] "library_source"        "library_strategy"      "molecule_ch1"         </span></span>
<span><span class="co">#&gt; [19] "organism_ch1"          "platform_id"           "relation"             </span></span>
<span><span class="co">#&gt; [22] "series_id"             "singlecellprobability" "source_name_ch1"      </span></span>
<span><span class="co">#&gt; [25] "sra_id"                "status"                "submission_date"      </span></span>
<span><span class="co">#&gt; [28] "taxid_ch1"             "title"                 "type"                 </span></span>
<span><span class="co">#&gt; [31] "Cor2CNT"               "HasAssignedCNT"        "CNTname"</span></span>
<span></span>
<span><span class="co">#Available gene annotations:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">orthosDB</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "seqnames"                     "start"                       </span></span>
<span><span class="co">#&gt;  [3] "end"                          "width"                       </span></span>
<span><span class="co">#&gt;  [5] "strand"                       "ENSEMBL_GENE_ID"             </span></span>
<span><span class="co">#&gt;  [7] "gene_name"                    "gene_biotype"                </span></span>
<span><span class="co">#&gt;  [9] "seq_coord_system"             "description"                 </span></span>
<span><span class="co">#&gt; [11] "ENSEMBL_GENE_ID_VERSION"      "ENSEMBL_CANONICAL_TRANSCRIPT"</span></span>
<span><span class="co">#&gt; [13] "GENE_SYMBOL"                  "ENTREZ_GENE_ID"              </span></span>
<span><span class="co">#&gt; [15] "ARCHS4_ID"</span></span>
<span></span>
<span><span class="co">#Retrieve partial annotation for a specific contrast</span></span>
<span><span class="co">#returned as a top-hit in the mouse caMKL1 query above:</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">orthosDB</span><span class="op">)</span><span class="op">[</span><span class="st">"GSM5021181"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"title"</span>, <span class="st">"series_id"</span>, <span class="st">"CNTname"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt; DataFrame with 1 row and 3 columns</span></span>
<span><span class="co">#&gt;                             title   series_id     CNTname</span></span>
<span><span class="co">#&gt;                       &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; GSM5021181 E0771_pRetrox_MRTFA_..   GSE164860  GSM5021179</span></span>
<span></span>
<span><span class="co"># Compare context and individual contrast fractions between</span></span>
<span><span class="co"># the mouse caMKL1 contrast under consideration and the "GSM5021181"</span></span>
<span><span class="co"># query hit:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">queryID</span> <span class="op">&lt;-</span> <span class="st">"GSM5021181"</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">contrast</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">dec_MKL1_mouse</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">unit</span> <span class="op">&lt;-</span> <span class="st">"L2FC"</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">contrast</span> <span class="op">==</span> <span class="st">"CONTEXT"</span><span class="op">)</span> <span class="op">{</span><span class="va">unit</span> <span class="op">&lt;-</span> <span class="st">"L2CPM"</span><span class="op">}</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">dec_MKL1_mouse</span><span class="op">)</span><span class="op">[[</span><span class="va">contrast</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>, <span class="st">"caMKL1"</span><span class="op">]</span>,</span>
<span>         <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">orthosDB</span><span class="op">)</span><span class="op">[[</span><span class="va">contrast</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>, <span class="va">queryID</span><span class="op">]</span>,</span>
<span>         pch <span class="op">=</span> <span class="fl">16</span>, cex <span class="op">=</span> <span class="fl">0.5</span>, col <span class="op">=</span> <span class="st">"darkslategrey"</span>, main <span class="op">=</span> <span class="va">contrast</span>,</span>
<span>         xlab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">unit</span>, <span class="st">" caMKL1"</span><span class="op">)</span>, ylab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">unit</span>, <span class="st">" "</span>, <span class="va">queryID</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, col <span class="op">=</span> <span class="st">"darkred"</span>, lwd <span class="op">=</span> <span class="fl">0.8</span>, lty <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="op">}</span>     </span></code></pre></div>
<p><img src="orthosIntro_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="advanced-use-cases-directly-accessing-the-orthos-models">Advanced use cases: Directly accessing the orthos models<a class="anchor" aria-label="anchor" href="#advanced-use-cases-directly-accessing-the-orthos-models"></a>
</h2>
<p>As (1) typical <code>orthos</code> use cases do not require direct
access to the models and (2) use of the models requires loading of a
conda environment via <code>basilisk</code> this functionality is by
default not exposed to the user and is carried out transparently by the
non-exported functions <code>.predictEncoder()</code> and
<code>.predictEncoderD()</code>.</p>
<p>However, as we envision cases where directly accessing the models
might be of interest we provide here a brief overview and examples for
direct calls to these functions.</p>
<p>The <code>orthos</code> models are implemented in <code>Keras</code>.
For each organism there are two types of models:</p>
<ul>
<li>A <strong>context encoder</strong> that produces a latent embedding
for a specific input context (represented as L2CPMs) and</li>
<li>A <strong>contrast conditional variational autoencoder</strong> that
first produces latent embeddings of input contrasts (represented as
LFCs) conditioned on context embeddings and then generates decoded
versions of those contrasts, again conditioned on the context embeddings
(see also figure in <a href="#prelude-a-short-overview-of-the-orthos-models">Prelude: A short
overview of the <code>orthos</code> models</a>).</li>
</ul>
<p>As noted previously the first time these models are requested either
by <code>decomposeVar</code> or directly by
<code>.predictEncoder()</code> and <code>.predictEncoderD()</code> they
are downloaded and cached in the user ExperimentHub directory (see
<code>ExperimentHub::getExperimentHubOption("CACHE")</code>) using the
<code>orthosData</code> companion package.</p>
<p>When calling the <code>.predictEncoder()</code> and
<code>.predictEncoderD()</code> methods directly be attentive to the
following:</p>
<ul>
<li>Before the call the predefined conda environment
<code>orthos:::orthosenv</code> needs to be activated using
<code><a href="https://rdrr.io/pkg/basilisk/man/basiliskStart.html" class="external-link">basilisk::basiliskStart()</a></code>
</li>
<li>The inputs passed to the models need to have appropriate size and
value representation (see examples below).</li>
<li>After the calls it’s good practice to deactivate the environment
with <code><a href="https://rdrr.io/pkg/basilisk/man/basiliskStart.html" class="external-link">basilisk::basiliskStop()</a></code>.</li>
</ul>
<p>We now demonstrate calls to the context encoder for generating a
latent embedding of a specific context and to the contrast conditional
variational autoencoder for producing a contrast latent embedding and
decoding.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># mouse MKL1 context and contrast with the appropriate shape and representation.</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># Shape of models input is M x N,  </span></span>
<span><span class="co"># where M is the number of conditions,</span></span>
<span><span class="co"># N the number of features -i.e orthos Genes</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># Representation is L2CPMs for contexts and L2FCs for contrasts.</span></span>
<span><span class="co">#</span></span>
<span><span class="va">CONTEXT</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">dec_MKL1_mouse</span>,<span class="st">"CONTEXT"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">CONTRAST</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">dec_MKL1_mouse</span>,<span class="st">"INPUT_CONTRASTS"</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Activate the `basilisk` environment:</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">basilisk</span><span class="op">)</span></span>
<span><span class="va">cl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/basilisk/man/basiliskStart.html" class="external-link">basiliskStart</a></span><span class="op">(</span><span class="fu">orthos</span><span class="fu">:::</span><span class="va">orthosenv</span>,</span>
<span>                    testload <span class="op">=</span> <span class="st">"tensorflow"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Produce a latent embedding for the context with .predictEncoder:</span></span>
<span><span class="va">LATC</span> <span class="op">&lt;-</span> <span class="fu">basilisk</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/basilisk/man/basiliskStart.html" class="external-link">basiliskRun</a></span><span class="op">(</span>proc <span class="op">=</span> <span class="va">cl</span>,</span>
<span>                              fun <span class="op">=</span> <span class="fu">orthos</span><span class="fu">:::</span><span class="va">.predictEncoder</span>,</span>
<span>                              organism <span class="op">=</span> <span class="st">"Mouse"</span>,</span>
<span>                              gene_input <span class="op">=</span> <span class="va">CONTEXT</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Produce a latent embedding and decoding for the contrast with .predictEncoderD:</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">basilisk</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/basilisk/man/basiliskStart.html" class="external-link">basiliskRun</a></span><span class="op">(</span>proc <span class="op">=</span> <span class="va">cl</span>,</span>
<span>                             fun <span class="op">=</span> <span class="fu">orthos</span><span class="fu">:::</span><span class="va">.predictEncoderD</span>,</span>
<span>                             organism <span class="op">=</span> <span class="st">"Mouse"</span>,</span>
<span>                             delta_input <span class="op">=</span> <span class="va">CONTRAST</span>, context <span class="op">=</span> <span class="va">LATC</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Deactivate the `basilisk` environment:</span></span>
<span><span class="fu">basilisk</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/basilisk/man/basiliskStart.html" class="external-link">basiliskStop</a></span><span class="op">(</span><span class="va">cl</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Access the contrast latent embedding and decoding from the .predictEncoderD returned result:</span></span>
<span><span class="va">LATD</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">$</span><span class="va">LATD</span></span>
<span><span class="va">DEC</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">$</span><span class="va">DEC</span></span></code></pre></div>
<p>Calls similar to the ones above are carried out under the hood when
<code><a href="../reference/decomposeVar.html">decomposeVar()</a></code> is called.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS Monterey 12.6.5</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats    graphics utils    stats4   methods  base    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] basilisk_1.12.0             ggplot2_3.4.2              </span></span>
<span><span class="co">#&gt;  [3] keras_2.11.1                orthosData_0.99.0          </span></span>
<span><span class="co">#&gt;  [5] orthos_0.99.0               SummarizedExperiment_1.30.1</span></span>
<span><span class="co">#&gt;  [7] Biobase_2.60.0              GenomicRanges_1.52.0       </span></span>
<span><span class="co">#&gt;  [9] GenomeInfoDb_1.36.0         IRanges_2.34.0             </span></span>
<span><span class="co">#&gt; [11] S4Vectors_0.38.1            BiocGenerics_0.46.0        </span></span>
<span><span class="co">#&gt; [13] MatrixGenerics_1.12.0       matrixStats_0.63.0         </span></span>
<span><span class="co">#&gt; [15] knitr_1.43                  BiocStyle_2.28.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] jsonlite_1.8.4                magrittr_2.0.3               </span></span>
<span><span class="co">#&gt;   [3] farver_2.1.1                  rmarkdown_2.21               </span></span>
<span><span class="co">#&gt;   [5] fs_1.6.2                      zlibbioc_1.46.0              </span></span>
<span><span class="co">#&gt;   [7] ragg_1.2.5                    vctrs_0.6.2                  </span></span>
<span><span class="co">#&gt;   [9] memoise_2.0.1                 RCurl_1.98-1.12              </span></span>
<span><span class="co">#&gt;  [11] base64enc_0.1-3               rstatix_0.7.2                </span></span>
<span><span class="co">#&gt;  [13] htmltools_0.5.5               S4Arrays_1.0.4               </span></span>
<span><span class="co">#&gt;  [15] AnnotationHub_3.8.0           curl_5.0.0                   </span></span>
<span><span class="co">#&gt;  [17] broom_1.0.4                   Rhdf5lib_1.22.0              </span></span>
<span><span class="co">#&gt;  [19] rhdf5_2.44.0                  sass_0.4.6                   </span></span>
<span><span class="co">#&gt;  [21] bslib_0.4.2                   desc_1.4.2                   </span></span>
<span><span class="co">#&gt;  [23] fontawesome_0.5.1             plyr_1.8.8                   </span></span>
<span><span class="co">#&gt;  [25] cachem_1.0.8                  whisker_0.4.1                </span></span>
<span><span class="co">#&gt;  [27] mime_0.12                     lifecycle_1.0.3              </span></span>
<span><span class="co">#&gt;  [29] pkgconfig_2.0.3               Matrix_1.5-4.1               </span></span>
<span><span class="co">#&gt;  [31] R6_2.5.1                      fastmap_1.1.1                </span></span>
<span><span class="co">#&gt;  [33] GenomeInfoDbData_1.2.10       shiny_1.7.4                  </span></span>
<span><span class="co">#&gt;  [35] digest_0.6.31                 colorspace_2.1-0             </span></span>
<span><span class="co">#&gt;  [37] AnnotationDbi_1.62.1          rprojroot_2.0.3              </span></span>
<span><span class="co">#&gt;  [39] ExperimentHub_2.8.0           textshaping_0.3.6            </span></span>
<span><span class="co">#&gt;  [41] RSQLite_2.3.1                 ggpubr_0.6.0                 </span></span>
<span><span class="co">#&gt;  [43] labeling_0.4.2                filelock_1.0.2               </span></span>
<span><span class="co">#&gt;  [45] fansi_1.0.4                   tfruns_1.5.1                 </span></span>
<span><span class="co">#&gt;  [47] httr_1.4.6                    abind_1.4-5                  </span></span>
<span><span class="co">#&gt;  [49] compiler_4.3.0                here_1.0.1                   </span></span>
<span><span class="co">#&gt;  [51] withr_2.5.0                   bit64_4.0.5                  </span></span>
<span><span class="co">#&gt;  [53] backports_1.4.1               BiocParallel_1.34.2          </span></span>
<span><span class="co">#&gt;  [55] carData_3.0-5                 DBI_1.1.3                    </span></span>
<span><span class="co">#&gt;  [57] highr_0.10                    HDF5Array_1.28.1             </span></span>
<span><span class="co">#&gt;  [59] tensorflow_2.11.0             ggsignif_0.6.4               </span></span>
<span><span class="co">#&gt;  [61] rappdirs_0.3.3                DelayedArray_0.26.3          </span></span>
<span><span class="co">#&gt;  [63] ggsci_3.0.0                   tools_4.3.0                  </span></span>
<span><span class="co">#&gt;  [65] interactiveDisplayBase_1.38.0 httpuv_1.6.11                </span></span>
<span><span class="co">#&gt;  [67] glue_1.6.2                    rhdf5filters_1.12.1          </span></span>
<span><span class="co">#&gt;  [69] promises_1.2.0.1              grid_4.3.0                   </span></span>
<span><span class="co">#&gt;  [71] generics_0.1.3                gtable_0.3.3                 </span></span>
<span><span class="co">#&gt;  [73] tidyr_1.3.0                   car_3.1-2                    </span></span>
<span><span class="co">#&gt;  [75] utf8_1.2.3                    XVector_0.40.0               </span></span>
<span><span class="co">#&gt;  [77] ggrepel_0.9.3                 BiocVersion_3.17.1           </span></span>
<span><span class="co">#&gt;  [79] pillar_1.9.0                  stringr_1.5.0                </span></span>
<span><span class="co">#&gt;  [81] later_1.3.1                   dplyr_1.1.2                  </span></span>
<span><span class="co">#&gt;  [83] BiocFileCache_2.8.0           lattice_0.21-8               </span></span>
<span><span class="co">#&gt;  [85] bit_4.0.5                     tidyselect_1.2.0             </span></span>
<span><span class="co">#&gt;  [87] Biostrings_2.68.1             bookdown_0.34                </span></span>
<span><span class="co">#&gt;  [89] xfun_0.39                     datasets_4.3.0               </span></span>
<span><span class="co">#&gt;  [91] stringi_1.7.12                yaml_2.3.7                   </span></span>
<span><span class="co">#&gt;  [93] evaluate_0.21                 codetools_0.2-19             </span></span>
<span><span class="co">#&gt;  [95] tibble_3.2.1                  BiocManager_1.30.20          </span></span>
<span><span class="co">#&gt;  [97] cli_3.6.1                     xtable_1.8-4                 </span></span>
<span><span class="co">#&gt;  [99] reticulate_1.28               systemfonts_1.0.4            </span></span>
<span><span class="co">#&gt; [101] munsell_0.5.0                 jquerylib_0.1.4              </span></span>
<span><span class="co">#&gt; [103] Rcpp_1.0.10                   dir.expiry_1.8.0             </span></span>
<span><span class="co">#&gt; [105] zeallot_0.1.0                 dbplyr_2.3.2                 </span></span>
<span><span class="co">#&gt; [107] png_0.1-8                     parallel_4.3.0               </span></span>
<span><span class="co">#&gt; [109] ellipsis_0.3.2                pkgdown_2.0.7                </span></span>
<span><span class="co">#&gt; [111] blob_1.2.4                    basilisk.utils_1.12.1        </span></span>
<span><span class="co">#&gt; [113] bitops_1.0-7                  viridisLite_0.4.2            </span></span>
<span><span class="co">#&gt; [115] scales_1.2.1                  grDevices_4.3.0              </span></span>
<span><span class="co">#&gt; [117] purrr_1.0.1                   crayon_1.5.2                 </span></span>
<span><span class="co">#&gt; [119] rlang_1.1.1                   cowplot_1.1.1                </span></span>
<span><span class="co">#&gt; [121] KEGGREST_1.40.0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-hu2019mkl1" class="csl-entry">
Hu, Xiao, Zongzhi Z Liu, Xinyue Chen, Vincent P Schulz, Abhishek Kumar,
Amaleah A Hartman, Jason Weinstein, et al. 2019. <span>“MKL1-Actin
Pathway Restricts Chromatin Accessibility and Prevents Mature
Pluripotency Activation.”</span> <em>Nature Communications</em> 10 (1):
1695.
</div>
<div id="ref-lachmann2018massive" class="csl-entry">
Lachmann, Alexander, Denis Torre, Alexandra B Keenan, Kathleen M
Jagodnik, Hoyjin J Lee, Lily Wang, Moshe C Silverstein, and Avi Ma’ayan.
2018. <span>“Massive Mining of Publicly Available RNA-Seq Data from
Human and Mouse.”</span> <em>Nature Communications</em> 9 (1): 1366.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Panagiotis Papasaikas, <a href="https://csoneson.github.io" class="external-link">Charlotte Soneson</a>, <a href="https://www.fmi.ch/platforms/platform.html?plt=115" class="external-link">Michael Stadler</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
